<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Zhuangzhuang Geng" />
  <meta name="author" content="Eric Wafula" />
  <meta name="author" content="Ryan J. Corbett" />
  <meta name="author" content="Yuanchao Zhang" />
  <meta name="author" content="Run Jin" />
  <meta name="author" content="Krutika S. Gaonkar" />
  <meta name="author" content="Sangeeta Shukla" />
  <meta name="author" content="Komal S. Rathi" />
  <meta name="author" content="Dave Hill" />
  <meta name="author" content="Aditya Lahiri" />
  <meta name="author" content="Daniel P. Miller" />
  <meta name="author" content="Alex Sickler" />
  <meta name="author" content="Kelsey Keith" />
  <meta name="author" content="Christopher Blackden" />
  <meta name="author" content="Antonia Chroni" />
  <meta name="author" content="Miguel A. Brown" />
  <meta name="author" content="Adam A. Kraya" />
  <meta name="author" content="Carl J. Koschmann" />
  <meta name="author" content="Kenneth Aldape" />
  <meta name="author" content="Xiaoyan Huang" />
  <meta name="author" content="Brian R. Rood" />
  <meta name="author" content="Jennifer L. Mason" />
  <meta name="author" content="Gerri R. Trooskin" />
  <meta name="author" content="Zied Abdullaev" />
  <meta name="author" content="Pei Wang" />
  <meta name="author" content="Yuankun Zhu" />
  <meta name="author" content="Bailey K. Farrow" />
  <meta name="author" content="Alvin Farrel" />
  <meta name="author" content="Joseph M. Dybas" />
  <meta name="author" content="Chuwei Zhong" />
  <meta name="author" content="Nicholas Van Kuren" />
  <meta name="author" content="Bo Zhang" />
  <meta name="author" content="Mariarita Santi" />
  <meta name="author" content="Saksham Phul" />
  <meta name="author" content="Asif T Chinwalla" />
  <meta name="author" content="Adam C. Resnick" />
  <meta name="author" content="Sharon J. Diskin" />
  <meta name="author" content="Sarah Tasian" />
  <meta name="author" content="Stephanie Stefankiewicz" />
  <meta name="author" content="John M. Maris" />
  <meta name="author" content="Brian M. Ennis" />
  <meta name="author" content="Matthew R. Lueder" />
  <meta name="author" content="Ammar S. Naqvi" />
  <meta name="author" content="Noel Coleman" />
  <meta name="author" content="Weiping Ma" />
  <meta name="author" content="Deanne Taylor" />
  <meta name="author" content="Jo Lynne Rokita" />
  <meta name="dcterms.date" content="2024-09-20" />
  <meta name="keywords" content="pediatric cancer, reproducibility, open science, multi-omics" />
  <title>The Open Pediatric Cancer Project</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="The Open Pediatric Cancer Project" />
  <meta name="citation_title" content="The Open Pediatric Cancer Project" />
  <meta property="og:title" content="The Open Pediatric Cancer Project" />
  <meta property="twitter:title" content="The Open Pediatric Cancer Project" />
  <meta name="dc.date" content="2024-09-20" />
  <meta name="citation_publication_date" content="2024-09-20" />
  <meta property="article:published_time" content="2024-09-20" />
  <meta name="dc.modified" content="2024-09-20T18:30:01+00:00" />
  <meta property="article:modified_time" content="2024-09-20T18:30:01+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Zhuangzhuang Geng" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0009-0007-6883-0691" />
  <meta name="citation_author" content="Eric Wafula" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-8073-3797" />
  <meta name="citation_author" content="Ryan J. Corbett" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-3478-0784" />
  <meta name="citation_author" content="Yuanchao Zhang" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="Run Jin" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-8958-9266" />
  <meta name="twitter:creator" content="@runjin" />
  <meta name="citation_author" content="Krutika S. Gaonkar" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0003-0838-2405" />
  <meta name="twitter:creator" content="@aggokittu" />
  <meta name="citation_author" content="Sangeeta Shukla" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-3727-9602" />
  <meta name="citation_author" content="Komal S. Rathi" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-5534-6904" />
  <meta name="twitter:creator" content="@komalsrathi" />
  <meta name="citation_author" content="Dave Hill" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-1337-1789" />
  <meta name="citation_author" content="Aditya Lahiri" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-9352-1312" />
  <meta name="citation_author" content="Daniel P. Miller" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-2032-4358" />
  <meta name="citation_author" content="Alex Sickler" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-7830-7537" />
  <meta name="citation_author" content="Kelsey Keith" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-7451-5117" />
  <meta name="citation_author" content="Christopher Blackden" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-4928-090X" />
  <meta name="citation_author" content="Antonia Chroni" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="Miguel A. Brown" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-6782-1442" />
  <meta name="twitter:creator" content="@migbro" />
  <meta name="citation_author" content="Adam A. Kraya" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-8526-5694" />
  <meta name="citation_author" content="Carl J. Koschmann" />
  <meta name="citation_author_institution" content="Department of Pediatrics, University of Michigan Health, Ann Arbor, MI, 48105, USA" />
  <meta name="citation_author_institution" content="Pediatric Hematology Oncology, Mott Children&#39;s Hospital, Ann Arbor, MI, 48109, USA" />
  <meta name="citation_author_orcid" content="0000-0002-0825-7615" />
  <meta name="citation_author" content="Kenneth Aldape" />
  <meta name="citation_author_institution" content="Laboratory of Pathology, National Cancer Institute, Bethesda, MD, 20892, USA" />
  <meta name="citation_author" content="Xiaoyan Huang" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-7267-4512" />
  <meta name="citation_author" content="Brian R. Rood" />
  <meta name="citation_author_institution" content="Children&#39;s National Research Institute, Washington, D.C." />
  <meta name="citation_author_institution" content="George Washington University School of Medicine and Health Sciences, Washington, D.C., 20052, USA" />
  <meta name="citation_author" content="Jennifer L. Mason" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="twitter:creator" content="@jenn0307" />
  <meta name="citation_author" content="Gerri R. Trooskin" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="Zied Abdullaev" />
  <meta name="citation_author_institution" content="Laboratory of Pathology, National Cancer Institute, Bethesda, MD, 20892, USA" />
  <meta name="citation_author" content="Pei Wang" />
  <meta name="citation_author_institution" content="Department of Genetics and Genomic Sciences, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA" />
  <meta name="citation_author_institution" content="Tisch Cancer Institute, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA" />
  <meta name="citation_author" content="Yuankun Zhu" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-2455-9525" />
  <meta name="twitter:creator" content="@zhuyuankun" />
  <meta name="citation_author" content="Bailey K. Farrow" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-6727-6333" />
  <meta name="citation_author" content="Alvin Farrel" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Oncology, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Center for Childhood Cancer Research, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0003-1087-9840" />
  <meta name="citation_author" content="Joseph M. Dybas" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="Chuwei Zhong" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0003-2406-2735" />
  <meta name="citation_author" content="Nicholas Van Kuren" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-7414-9516" />
  <meta name="citation_author" content="Bo Zhang" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-0743-5379" />
  <meta name="citation_author" content="Mariarita Santi" />
  <meta name="citation_author_institution" content="Department of Pathology and Laboratory Medicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Pathology and Laboratory Medicine, University of Pennsylvania Perelman School of Medicine, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-6728-3450" />
  <meta name="citation_author" content="Saksham Phul" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-2771-2572" />
  <meta name="twitter:creator" content="@migbro" />
  <meta name="citation_author" content="Asif T Chinwalla" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-7831-3996" />
  <meta name="citation_author" content="Adam C. Resnick" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0003-0436-4189" />
  <meta name="twitter:creator" content="@adamcresnick" />
  <meta name="citation_author" content="Sharon J. Diskin" />
  <meta name="citation_author_institution" content="Division of Oncology, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Pediatrics, University of Pennsylvania, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-7200-8939" />
  <meta name="twitter:creator" content="@sjdiskin" />
  <meta name="citation_author" content="Sarah Tasian" />
  <meta name="citation_author_institution" content="Division of Oncology, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Pediatrics, University of Pennsylvania, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="Stephanie Stefankiewicz" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="John M. Maris" />
  <meta name="citation_author_institution" content="Division of Oncology, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Pediatrics, University of Pennsylvania, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author" content="Brian M. Ennis" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-2653-5009" />
  <meta name="citation_author" content="Matthew R. Lueder" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Pathology and Laboratory Medicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0009-0002-7370-102X" />
  <meta name="citation_author" content="Ammar S. Naqvi" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="twitter:creator" content="@jenn0307" />
  <meta name="citation_author" content="Noel Coleman" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0001-6454-1285" />
  <meta name="citation_author" content="Weiping Ma" />
  <meta name="citation_author_institution" content="Department of Genetics and Genomic Sciences, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA" />
  <meta name="citation_author_institution" content="Tisch Cancer Institute, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA" />
  <meta name="citation_author" content="Deanne Taylor" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Pediatrics, University of Pennsylvania Perelman Medical School, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-3302-4610" />
  <meta name="citation_author" content="Jo Lynne Rokita" />
  <meta name="citation_author_institution" content="Center for Data-Driven Discovery in Biomedicine, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Division of Neurosurgery, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_institution" content="Department of Biomedical and Health Informatics, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0003-2171-3627" />
  <meta name="twitter:creator" content="@jolynnerokita" />
  <link rel="canonical" href="https://rokitalab.github.io/OPC-manuscript/" />
  <meta property="og:url" content="https://rokitalab.github.io/OPC-manuscript/" />
  <meta property="twitter:url" content="https://rokitalab.github.io/OPC-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://rokitalab.github.io/OPC-manuscript/" />
  <meta name="citation_pdf_url" content="https://rokitalab.github.io/OPC-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://rokitalab.github.io/OPC-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://rokitalab.github.io/OPC-manuscript/v/d444a1fad5b0c9bb1a1429eec2a0e05c54a3dc80/" />
  <meta name="manubot_html_url_versioned" content="https://rokitalab.github.io/OPC-manuscript/v/d444a1fad5b0c9bb1a1429eec2a0e05c54a3dc80/" />
  <meta name="manubot_pdf_url_versioned" content="https://rokitalab.github.io/OPC-manuscript/v/d444a1fad5b0c9bb1a1429eec2a0e05c54a3dc80/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">The Open Pediatric Cancer Project</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://rokitalab.github.io/OPC-manuscript/v/d444a1fad5b0c9bb1a1429eec2a0e05c54a3dc80/">permalink</a>)
was automatically generated
from <a href="https://github.com/rokitalab/OPC-manuscript/tree/d444a1fad5b0c9bb1a1429eec2a0e05c54a3dc80">rokitalab/OPC-manuscript@d444a1f</a>
on 2024-09-20.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Zhuangzhuang Geng</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0009-0007-6883-0691">0009-0007-6883-0691</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/zzgeng">zzgeng</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Eric Wafula</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8073-3797">0000-0001-8073-3797</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/ewafula">ewafula</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Ryan J. Corbett</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3478-0784">0000-0002-3478-0784</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/rjcorb">rjcorb</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Yuanchao Zhang</strong><br><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Run Jin</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8958-9266">0000-0002-8958-9266</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/runjin326">runjin326</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Krutika S. Gaonkar</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0838-2405">0000-0003-0838-2405</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Sangeeta Shukla</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3727-9602">0000-0002-3727-9602</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/sangeetashukla">sangeetashukla</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Komal S. Rathi</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-5534-6904">0000-0001-5534-6904</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/komalsrathi">komalsrathi</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Dave Hill</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1337-1789">0000-0002-1337-1789</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/atlas4213">atlas4213</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Aditya Lahiri</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9352-1312">0000-0001-9352-1312</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/adilahiri">adilahiri</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Daniel P. Miller</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2032-4358">0000-0002-2032-4358</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/dmiller15">dmiller15</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Alex Sickler</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7830-7537">0000-0001-7830-7537</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/sickler-alex">sickler-alex</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Kelsey Keith</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7451-5117">0000-0002-7451-5117</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/kelseykeith">kelseykeith</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Christopher Blackden</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4928-090X">0000-0002-4928-090X</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/devbyaccident">devbyaccident</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Antonia Chroni</strong><br>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/AntoniaChroni">AntoniaChroni</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Miguel A. Brown</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6782-1442">0000-0001-6782-1442</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/migbro">migbro</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Adam A. Kraya</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8526-5694">0000-0002-8526-5694</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/aadamk">aadamk</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Carl J. Koschmann</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0825-7615">0000-0002-0825-7615</a><br>
<small>
Department of Pediatrics, University of Michigan Health, Ann Arbor, MI, 48105, USA; Pediatric Hematology Oncology, Mott Children’s Hospital, Ann Arbor, MI, 48109, USA
</small></p></li>
<li><p><strong>Kenneth Aldape</strong><br><br>
<small>
Laboratory of Pathology, National Cancer Institute, Bethesda, MD, 20892, USA
</small></p></li>
<li><p><strong>Xiaoyan Huang</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7267-4512">0000-0001-7267-4512</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/HuangXiaoyan0106">HuangXiaoyan0106</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Brian R. Rood</strong><br><br>
<small>
Children’s National Research Institute, Washington, D.C.; George Washington University School of Medicine and Health Sciences, Washington, D.C., 20052, USA
</small></p></li>
<li><p><strong>Jennifer L. Mason</strong><br><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Gerri R. Trooskin</strong><br><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Zied Abdullaev</strong><br><br>
<small>
Laboratory of Pathology, National Cancer Institute, Bethesda, MD, 20892, USA
</small></p></li>
<li><p><strong>Pei Wang</strong><br><br>
<small>
Department of Genetics and Genomic Sciences, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA; Tisch Cancer Institute, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA
</small></p></li>
<li><p><strong>Yuankun Zhu</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2455-9525">0000-0002-2455-9525</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/yuankunzhu">yuankunzhu</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Bailey K. Farrow</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6727-6333">0000-0001-6727-6333</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/baileyckelly">baileyckelly</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Alvin Farrel</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1087-9840">0000-0003-1087-9840</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/afarrel">afarrel</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Oncology, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Center for Childhood Cancer Research, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
· Funded by NCI/NIH Contract No. 75N91019D00024, Task Order No. 75N91020F00003
</small></p></li>
<li><p><strong>Joseph M. Dybas</strong><br>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/JosephDybas">JosephDybas</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Chuwei Zhong</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-2406-2735">0000-0003-2406-2735</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/zoomzoom1011">zoomzoom1011</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Nicholas Van Kuren</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7414-9516">0000-0002-7414-9516</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/nicholasvk">nicholasvk</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Bo Zhang</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0743-5379">0000-0002-0743-5379</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/zhangb1">zhangb1</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Mariarita Santi</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6728-3450">0000-0002-6728-3450</a><br>
<small>
Department of Pathology and Laboratory Medicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Pathology and Laboratory Medicine, University of Pennsylvania Perelman School of Medicine, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Saksham Phul</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2771-2572">0000-0002-2771-2572</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/sakshamphul">sakshamphul</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Asif T Chinwalla</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7831-3996">0000-0001-7831-3996</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/chinwallaa">chinwallaa</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Adam C. Resnick</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0436-4189">0000-0003-0436-4189</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/adamcresnick">adamcresnick</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
· Funded by Children’s Brain Tumor Network; NIH 3P30 CA016520-44S5, U2C HL138346-03, U24 CA220457-03; NCI/NIH Contract No. 75N91019D00024, Task Order No. 75N91020F00003; Children’s Hospital of Philadelphia Division of Neurosurgery
</small></p></li>
<li><p><strong>Sharon J. Diskin</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7200-8939">0000-0002-7200-8939</a><br>
<small>
Division of Oncology, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Pediatrics, University of Pennsylvania, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Sarah Tasian</strong><br><br>
<small>
Division of Oncology, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Pediatrics, University of Pennsylvania, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Stephanie Stefankiewicz</strong><br><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>John M. Maris</strong><br><br>
<small>
Division of Oncology, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Pediatrics, University of Pennsylvania, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Brian M. Ennis</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2653-5009">0000-0002-2653-5009</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/bmennis">bmennis</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Matthew R. Lueder</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0009-0002-7370-102X">0009-0002-7370-102X</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/luederm">luederm</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Pathology and Laboratory Medicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Ammar S. Naqvi</strong><br>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/naqvia">naqvia</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Noel Coleman</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6454-1285">0000-0001-6454-1285</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
</small></p></li>
<li><p><strong>Weiping Ma</strong><br><br>
<small>
Department of Genetics and Genomic Sciences, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA; Tisch Cancer Institute, Icahn School of Medicine at Mount Sinai, New York, NY 10029, USA
</small></p></li>
<li><p><strong>Deanne Taylor</strong><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3302-4610">0000-0002-3302-4610</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/taylordm">taylordm</a><br>
<small>
Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Pediatrics, University of Pennsylvania Perelman Medical School, Philadelphia, PA, 19104, USA
· Funded by NCI/NIH Contract No. 75N91019D00024, Task Order No. 75N91020F00003
</small></p></li>
<li><p><strong>Jo Lynne Rokita</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.png" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-2171-3627">0000-0003-2171-3627</a>
· <img src="images/github.png" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/jharenza">jharenza</a><br>
<small>
Center for Data-Driven Discovery in Biomedicine, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Division of Neurosurgery, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA; Department of Biomedical and Health Informatics, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
· Funded by NCI/NIH Contract No. 75N91019D00024, Task Order No. 75N91020F00003
</small></p></li>
</ul>
<h2 id="contact-information">Contact information</h2>
<p>✉Correspondence: Jo Lynne Rokita <a href="mailto:rokita@chop.edu" class="email">rokita@chop.edu</a></p>
<!-- The title of the article, which should include an accurate, clear and concise description of the reported work, avoiding abbreviations; and the full names, institutional addresses, e-mail addresses for all authors. The corresponding author should also be indicated. It is suitable, should the authors of the work prefer, to designate authorship as a Consortium or Project. In such a case, a contact author must still be provided, and a complete author list/institutions/emails/etc should be included in the Authors' Contribution section. -->
<h2 id="abstract">Abstract</h2>
<!-- 250 words maximum for abstract. Abstract must be structured into separate sections: 1). Background; 2). Findings; 3). Conclusions. -->
<p><strong>Background:</strong> In 2019, the Open Pediatric Brain Tumor Atlas (OpenPBTA) was created as a global, collaborative open-science initiative to genomically characterize 1,074 pediatric brain tumors and 22 patient-derived cell lines.
Here, we extend the OpenPBTA to create the Open Pediatric Cancer (OpenPedCan) Project, a harmonized open-source multi-omic dataset from 6,112 pediatric cancer patients with 7,096 tumor events across more than 100 histologies.
Combined with RNA-Seq from the Genotype-Tissue Expression (GTEx) and The Cancer Genome Atlas (TCGA), OpenPedCan contains nearly 48,000 total biospecimens (24,002 tumor and 23,893 normal specimens).</p>
<!-- A very brief preview of the data type(s) produced, the methods used, and information relevant to data validation -->
<p><strong>Findings:</strong> We utilized Gabriella Miller Kids First (GMKF) workflows to harmonize WGS, WXS, RNA-seq, and Targeted Sequencing datasets to include somatic SNVs, InDels, CNVs, SVs, RNA expression, fusions, and splice variants.
We integrated summarized CPTAC whole cell proteomics and phospho-proteomics data, miRNA-Seq data, and have developed a methylation array harmonization workflow to include m-values, beta-vales, and copy number calls.
OpenPedCan contains reproducible, dockerized workflows in GitHub, CAVATICA, and Amazon Web Services (AWS) to deliver harmonized and processed data from over 60 scalable modules which can be leveraged both locally and on AWS.
The processed data are released in a versioned manner and accessible through CAVATICA or AWS S3 download (from GitHub), and queryable through PedcBioPortal and the NCI’s pediatric Molecular Targets Platform.
Notably, we have expanded PBTA molecular subtyping to include methylation information to align with the WHO 2021 Central Nervous System Tumor classifications, allowing us to create research-grade integrated diagnoses for these tumors.</p>
<!-- A short summary of the potential uses of these data and implications for the field. -->
<p><strong>Conclusions:</strong> OpenPedCan data and its reproducible analysis module framework are openly available and can be utilized and/or adapted by researchers to accelerate discovery, validation, and clinical translation.</p>
<h2 id="keywords">Keywords</h2>
<p>Pediatric cancer, open science, reproducibility, multi-omics</p>
<h2 id="data-description">Data Description</h2>
<!-- A brief statement providing background and purpose for collection of these data should be presented for readers without specialist knowledge in that area. A clear, concise, description of the data, the protocol(s) for data collection, data curation and quality control, as well as potential uses should then follow. -->
<p>The Open Pediatric Cancer (OpenPedCan) project at the Children’s Hospital of Philadelphia (CHOP) is an open analysis effort in which we harmonize pediatric cancer data from multiple sources, perform downstream cancer analyses on these data, and provide them on PedcBioPortal and v2.1 of NCI’s <a href="https://moleculartargets.ccdi.cancer.gov/">Pediatric Molecular Targets Platform (MTP)</a>.
We harmonized, aggregated, and analyzed data from multiple pediatric and adult data sources, building upon the work of the OpenPBTA (<strong>Figure <a href="#fig:Fig1">1</a></strong>).
Biospecimen-level metadata and clinical data are contained in <a href="https://github.com/d3b-center/OpenPedCan-analysis/blob/af9e2bb87e552159309c277ff8fef88c16fe8dcf/tables/results/SuppTable1-Histologies.xlsx"><strong>Supplemental Table 1</strong></a>.</p>
<div id="fig:Fig1" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/d3b-center/OpenPedCan-analysis/e0e35bb13fd8542b807f7ea75ffd3ab857c522cb/figures/manuscript_OPC/figure1/Figure1.png?sanitize=true" style="width:7in" alt="Figure 1: OpenPedCan Data. A, OpenPedCan contains multi-omic data from seven cohorts of pediatric tumors (A-B) with counts by tumor event, RNA-Seq from adult tumors from The Cancer Genome Atlas (TCGA) Program (C-D) and RNA-Seq from normal adult tissues from the Genotype-Tissue Expression (GTeX) project (E) with counts by specimen. (Abbreviations: TARGET = Therapeutically Applicable Research to Generate Effective Treatments , PPTC = Pediatric Preclinical Testing Consortium, PBTA = Pediatric Brain Tumor Atlas, Maris = Neuroblastoma cell lines from the Maris Laboratory at CHOP, GMKF = Gabriella Miller Kids First, DGD = Division of Genomic Diagnostics at CHOP, CPTAC = Clinical Proteomic Tumor Analysis Consortium)" />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>OpenPedCan Data.</strong> A, OpenPedCan contains multi-omic data from seven cohorts of pediatric tumors (A-B) with counts by tumor event, RNA-Seq from adult tumors from The Cancer Genome Atlas (TCGA) Program (C-D) and RNA-Seq from normal adult tissues from the Genotype-Tissue Expression (GTeX) project (E) with counts by specimen. (Abbreviations: TARGET = Therapeutically Applicable Research to Generate Effective Treatments , PPTC = Pediatric Preclinical Testing Consortium, PBTA = Pediatric Brain Tumor Atlas, Maris = Neuroblastoma cell lines from the Maris Laboratory at CHOP, GMKF = Gabriella Miller Kids First, DGD = Division of Genomic Diagnostics at CHOP, CPTAC = Clinical Proteomic Tumor Analysis Consortium)</figcaption>
</figure>
</div>
<p>OpenPedCan currently include the following datasets, described more fully below:</p>
<ul>
<li>OpenPBTA</li>
<li>TARGET</li>
<li>Kids First Neuroblastoma (X01)</li>
<li>Kids First PBTA (X01)</li>
<li>Chordoma Foundation</li>
<li>PPTC</li>
<li>Maris</li>
<li>MI-ONCOSEQ Study</li>
<li>DGD</li>
<li>GTEx</li>
<li>TCGA</li>
<li>CPTAC PBTA</li>
<li>CPTAC GBM</li>
<li>HOPE proteomics</li>
</ul>
<p><strong>Open Pediatric Brain Tumor Atlas (OpenPBTA)</strong></p>
<p>In September of 2018, the <a href="https://cbtn.org/">Children’s Brain Tumor Network (CBTN)</a> released the <a href="https://cbtn.org/pediatric-brain-tumor-atlas/">Pediatric Brain Tumor Atlas (PBTA)</a>, a genomic dataset (whole genome sequencing, whole exome sequencing, RNA sequencing, proteomic, and clinical data) for nearly 1,000 tumors, available from the <a href="https://kidsfirstdrc.org/">Gabriella Miller Kids First Portal</a>.
In September of 2019, the Open Pediatric Brain Tumor Atlas (OpenPBTA) Project was launched.
OpenPBTA was a global open science initiative to comprehensively define the molecular landscape of tumors of 943 patients from the CBTN and the PNOC003 DIPG clinical trial from the <a href="http://www.pnoc.us/">Pediatric Pacific Neuro-oncology Consortium</a> through real-time, collaborative analyses and collaborative manuscript writing on GitHub <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
Additional PBTA data has been, and will be continually added to, OpenPedCan.</p>
<p><strong>Therapeutically Applicable Research to Generate Effective Treatments <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000218.v23.p8">(TARGET)</a></strong></p>
<p>The Therapeutically Applicable Research to Generate Effective Treatments (TARGET) Initiative is an NCI-funded collection of disease-specific projects that seeks to identify the genomic changes of pediatric cancers.
The overall goal is to collect genomic data to accelerate the development of more effective therapies.
OpenPedCan analyses include the seven diseases present in the TARGET dataset: Acute Lymphoblastic Leukemia (ALL), Acute Myeloid Leukemia (AML), Clear cell sarcoma of the kidney, Neuroblastoma, Osteosarcoma, Rhabdoid tumor, and Wilm’s Tumor.</p>
<p><strong>Gabriella Miller Kids First <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001436.v1.p1">(Neuroblastoma)</a> and <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs002517.v2.p2">PBTA</a></strong></p>
<p>The Gabriella Miller Kids First Pediatric Research Program (Kids First) is a large-scale effort to accelerate research and gene discovery in pediatric cancers and structural birth defects.
The program includes whole genome sequencing (WGS) from patients with pediatric cancers and structural birth defects and their families.
OpenPedCan analyses include Neuroblastoma and PBTA data from the Kids First projects.</p>
<p><strong>Chordoma Foundation</strong></p>
<p>The <a href="https://www.chordomafoundation.org">Chordoma Foundation</a> seeks to advance research and improve healthcare for patients diagnosed with chordoma and has shared patient and model sequencing data with the CBTN.</p>
<p><strong>Pediatric Preclinical Testing Consortium <a href="@https://ctep.cancer.gov/MajorInitiatives/Pediatric_PIVOT_Program.htm%5D">(PPTC)</a></strong></p>
<p>The National Cancer Institute’s (NCI) former PPTC, now the <a href="https://preclinicalpivot.org/about-pivot/">Pediatric Preclinical in Vivo Testing (PIVOT) Program</a>, molecularly and pharmacologically characterizes cell-derived and patient-derived xenograft (PDX) models.
OpenPedCan includes re-harmonized RNA-Seq data for 244 models from the initial PPTC study <span class="citation" data-cites="SDwYl8uA">[<a href="#ref-SDwYl8uA" role="doc-biblioref">2</a>]</span>.</p>
<p><strong>MI-ONCOSEQ Study <span class="citation" data-cites="ygVj7a6q">[<a href="#ref-ygVj7a6q" role="doc-biblioref">3</a>]</span></strong></p>
<p>These clinical sequencing data from the University of Michigan were donated to CBTN and added to the PBTA cohort.</p>
<p><strong>Division of Genomic Diagnostics at Children’s Hospital of Philadelphia <a href="https://www.chop.edu/cancer-panels">(DGD)</a></strong></p>
<p>CHOP’s <a href="https://www.chop.edu/centers-programs/division-genomic-diagnostics">Division of Genomic Diagnostics</a> has partnered with CCDI to add somatic panel sequencing data to OpenPedCan and the Molecular Targets Platform.</p>
<p><strong>The Genotype-Tissue Expression Project <a href="https://gtexportal.org/home/">(GTEx)</a></strong></p>
<p>The GTEx project is an ongoing effort to build a comprehensive public data resource and tissue bank to study tissue-specific gene expression, regulation and their relationship with genetic variants.
Samples were collected from 54 non-diseased tissue sites across nearly 1000 individuals, primarily for molecular assays including WGS, WXS, and RNA-Seq.
OpenPedCan project includes 17,382 GTEx RNA-Seq samples from GTEx v8 release, which span across 31 GTEx groups in the v12 release.</p>
<p><strong>The Cancer Genome Atlas Program <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000178.v11.p8">(TCGA)</a></strong></p>
<p>TCGA is a landmark cancer genomics program that molecularly characterized over 20,000 primary cancer and matched normal samples spanning 33 cancer types.
It is a joint effort between NCI and the National Human Genome Research Institute.
OpenPedCan project includes 10,414 TCGA RNA-Seq samples (716 normal and 9698 tumor) from <a href="https://github.com/d3b-center/OpenPedCan-analysis/blob/dev/analyses/molecular-subtyping-integrate/results/tcga_cancer_groups.tsv">33 cancer types</a>.</p>
<p><strong>Clinical Proteomic Tumor Analysis Consortium (CPTAC) PBTA proteomics study</strong></p>
<p>The CPTAC pediatric pan-brain tumor study <span class="citation" data-cites="meH98mKZ">[<a href="#ref-meH98mKZ" role="doc-biblioref">4</a>]</span> contains 218 tumors profiled by proteogenomics and are included in OPC.</p>
<p><strong>CPTAC adult GBM proteomics study</strong></p>
<p>This CPTAC adult GBM study <span class="citation" data-cites="XOVss5Jf">[<a href="#ref-XOVss5Jf" role="doc-biblioref">5</a>]</span> contains 99 tumors profiled by proteogenomics and are included in OPC.</p>
<p><strong>Project HOPE proteomics study</strong></p>
<p>Project HOPE is an adolescent and young adult high-grade glioma study (in preparation for publication) that contains 90 tumors profiled by proteogenomics and are included in OPC.</p>
<h2 id="context">Context</h2>
<!-- Please state what motivated you to produced this data, and give some background on the area of research to relate the work to previous studies and help understand its utility and potential for reuse. -->
<p>Creation of this dataset had multiple motivations.
First, we sought to harmonize, summarize, and contextualize pediatric cancer genomics data among normal tissues (GTEx) and adult cancer tissues (TCGA) to enable the creation of the National Cancer Institute’s Molecular Targets Platform (MTP) at https://moleculartargets.ccdi.cancer.gov/.
Next, we created this resource for broad community use to promote rapid reuse and accelerate the discovery of additional mechanisms contributing to the pathogenesis of pediatric cancers and/or to identify novel candidate therapeutic targets for pediatric cancer.</p>
<p>Similar to OpenPBTA, OpenPedCan operates on a pull request model to accept contributions.
We set up continuous integration software via GitHub Actions to confirm the reproducibility of analyses within the project’s Docker container.
We maintained a data release folder on Amazon S3, downloadable directly from S3 or our open-access CAVATICA project, with merged files for each analysis.
As we produced new results, identified data issues, or added additional data, we created new data releases in a versioned manner.
The project maintainers include scientists from the <a href="https://d3b.center/">Center for Data-Driven Discovery in Biomedicine</a> and formerly the <a href="https://www.research.chop.edu/department-of-biomedical-and-health-informatics">Department of Biomedical and Health Informatics</a> at the Children’s Hospital of Philadelphia.</p>
<h2 id="methods">Methods</h2>
<p>An overview of the OpenPedCan methods is depicted in <strong>Figure <a href="#fig:Fig2">2</a></strong>.
Briefly, most primary harmonization analysis workflows were performed with Kids First pipelines written in Common Workflow Language (CWL) using CAVATICA (detailed below).
Alignment and expression quantification for GTEx and TCGA RNA-Seq was performed by the respective consortium.
Custom python, R, and/or bash scripts were then created in OpenPedCan using the primary harmonized output files.</p>
<div id="fig:Fig2" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/d3b-center/OpenPedCan-analysis/e0e35bb13fd8542b807f7ea75ffd3ab857c522cb/figures/manuscript_OPC/figure2/Figure2.png?sanitize=true" style="width:7in" alt="Figure 2: OpenPedCan Analysis Workflow. Depicted are the datasets (yellow, orange, and grey) contained within OpenPedCan. These datasets are made available in a harmonized manner through primary analysis workflows (blue) for DNA, RNA, and/or proteogenomics data. Files derived from the primary analysis workflows (green) are released within OpenPedCan. Additional analysis modules developed within OpenPedCan (red) also generate results files (green) which are released within OpenPedCan." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>OpenPedCan Analysis Workflow.</strong> Depicted are the datasets (yellow, orange, and grey) contained within OpenPedCan. These datasets are made available in a harmonized manner through primary analysis workflows (blue) for DNA, RNA, and/or proteogenomics data. Files derived from the primary analysis workflows (green) are released within OpenPedCan. Additional analysis modules developed within OpenPedCan (red) also generate results files (green) which are released within OpenPedCan.</figcaption>
</figure>
</div>
<!-- This section can potentially be further broken up into subheadings a) Sampling strategy and b) Steps. Related methods can also be grouped under corresponding subheadings. This section should provide enough detail to allow other researchers to interpret and repeat the study. In addition to including the design of the study, it needs to provide detail on the type of reagents, instrumentation and kits used (including any identifiers such as RRIDs).

Please include and cite the URLs (or DOIs if snapshots have been archived in a repository such as Zenodo or GigaDB) to any publically available bioinformatics tools that are used in the production of the data, including the exact version used. If you are using new bioinformatics tools that you have developed for production or processing of the data described here, this may be suitable for submission as a Technical Note. We strongly encourage the use of workflow management systems such as Galaxy and container systems such as Docker, to save the details of the methods to encourage reproducibility as well as conciseness. Our GigaDB repository can also be used to archive data, workflows and snapshots of the code with an accompanying DOI. We also offer integration with Code Ocean, a cloud-based computational reproducibility platform.

For any article that presents software, in addition to making the source code available, we encourage submission to Code Ocean, a cloud-based computational reproducibility platform. Code Ocean supports any open-source programming language (as well as MATLAB and Stata). Authors should upload their code and data, specifying the appropriate languages and dependencies within the computational environment, to create a “compute capsule” that reproduces the results.

Once the code is published on Code Ocean, you will be issued DOI, which should be used to cite the compute capsule in the software section and any other relevant reference in the paper. Readers will then be able to view and verify the article’s results without needing to download or install anything. To submit, sign up to CodeOcean.
-->
<h3 id="method-details">Method Details</h3>
<h4 id="nucleic-acids-extraction-and-library-preparation-pbta-x01-and-mirna-seq">Nucleic acids extraction and library preparation (PBTA X01 and miRNA-Seq)</h4>
<p>For detailed methods about the OpenPBTA cohort, please refer to the manuscript <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
For the PBTA X01 cohort, libraries were prepped using the Illumina TruSeq Strand-Specific Protocol to pull out poly-adenylated transcripts.</p>
<p><strong>cDNA Library Construction</strong></p>
<p>Total RNA was quantified using the Quant-iT™ RiboGreen® RNA Assay Kit and normalized to 5ng/ul.
Following plating, 2 uL of ERCC controls (using a 1:1000 dilution) were spiked into each sample.
An aliquot of 325 ng for each sample was transferred into library preparation.
The resultant 400bp cDNA went through dual-indexed library preparation: ‘A’ base addition, adapter ligation using P7 adapters, and PCR enrichment using P5 adapters.
After enrichment, the libraries were quantified using Quant-iT PicoGreen (1:200 dilution).
Samples were normalized to 5 ng/uL.
The sample set was pooled and quantified using the KAPA Library Quantification Kit for Illumina Sequencing Platforms.</p>
<p><strong>miRNA Extraction and Library Preparation</strong></p>
<p>Total RNA for CBTN samples was extracted as described in OpenPBTA <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span> and prepared according to the HTG Edge Seq protocol for the extracted RNA miRNA Whole transcriptome assay (WTA).
15ng of RNA were mixed in 25ul of lysis buffer, which were then loaded onto a 96-well plate.
Human Fetal Brain Total RNA (Takara Bio USA, #636526) and Human Brain Total RNA (Ambion, Inc., Austin, TX, USA) were used as controls.
The plate was loaded into the HTG EdgeSeq processor along with the miRNA WTA assay reagent pack.
Samples were processed for 18-20 hours, then were barcoded and amplified using a unique forward and reverse primer combination.
PCR settings used for barcoding and amplification were 95C for 4 min, 16 cycles of (95C for 15 sec, 56C for 45 sec, 68C for 45 sec), and 68C for 10 min.
Barcoded and amplified samples were cleaned using AMPure magnetic beads (Ampure XP,Cat# A63881).
Libraries were quantified using the KAPA Biosystem assay qPCR kit (Kapa Biosystems Cat#KK4824) and CT values were used to determine the pM concentration of each library.</p>
<h4 id="data-generation">Data generation</h4>
<p><strong>PBTA X01 Illumina Sequencing</strong>
Pooled libraries were normalized to 2nM and denatured using 0.1 N NaOH prior to sequencing.
Flowcell cluster amplification and sequencing were performed according to the manufacturer’s protocols using the NovaSeq 6000.
Each run was a 151bp paired-end with an eight-base index barcode read.
Data was analyzed using the Broad Picard Pipeline which includes de-multiplexing and data aggregation.</p>
<p><strong>PBTA miRNA Sequencing</strong>
Libraries were pooled, denatured, and loaded onto sequencing cartridge.
Libraries were sequenced using an Illumina Nextseq 500 per manufacturer guidelines.
FASTQ files were generated from raw sequencing data using Illumina BaseSpace and analyzed with the HTG EdgeSeq Parser software v5.4.0.7543 to generate an excel file containing quantification of 2083 miRNAs per sample.
Any sample that did not pass the quality control set by the HTG REVEAL software version 2.0.1 (Tuscon, AR, USA) was excluded from the analysis.</p>
<h4 id="dna-wgs-alignment-and-snp-calling">DNA WGS Alignment and SNP Calling</h4>
<p>Please refer to the OpenPBTA manuscript for details on DNA WGS Alignment, prediction of participants’ genetic sex, and SNP calling for B-allele Frequency (BAF) generation. <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.</p>
<h4 id="somatic-mutation-and-indel-calling">Somatic Mutation and INDEL Calling</h4>
<p>For matched tumor/normal samples, we used the same mutation calling methods as described in OpenPBTA manuscript for details <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
For tumor only samples, we ran Mutect2 from GATK v4.2.2.0 using the following <a href="https://github.com/kids-first/kf-tumor-workflow/tree/v0.3.0-beta">workflow</a>.</p>
<p><strong>VCF annotation and MAF creation</strong></p>
<p>Somatic variants were annotated by the Ensembl Variant Effect Predictor (VEP v105) <span class="citation" data-cites="p1f5DxRQ">[<a href="#ref-p1f5DxRQ" role="doc-biblioref">6</a>]</span>.
From tumor only variant calls, we removed variants with <code>alt_depth == 0</code> or <code>t_depth &lt; 4</code>.</p>
<p><strong>Consensus SNV Calling (tumor/normal only)</strong></p>
<p>We adopted the consensus SNV calling method described in OpenPBTA manuscript with adjustment <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
For SNV calling, we combined four consensus SNV calling algorithms: Strelka2<span class="citation" data-cites="REfkDUtE">[<a href="#ref-REfkDUtE" role="doc-biblioref">7</a>]</span>, Mutect2<span class="citation" data-cites="149BEKISi">[<a href="#ref-149BEKISi" role="doc-biblioref">8</a>]</span>, Lancet<span class="citation" data-cites="V6KdWVYi">[<a href="#ref-V6KdWVYi" role="doc-biblioref">9</a>]</span>, and VarDict<span class="citation" data-cites="trQRR8fs">[<a href="#ref-trQRR8fs" role="doc-biblioref">10</a>]</span>.</p>
<p>Strelka2 outputs multi-nucleotide polymorphisms (MNPs) as consecutive single-nucleotide polymorphisms.
In order preserve MNPs, we gather MNP calls from the other caller inputs, and search for evidence supporting these consecutive SNP calls as MNP candidates.
Once found, the Strelka2 SNP calls supporting a MNP are converted to a single MNP call.
This is done to preserve the predicted gene model as accurately as possible in our consensus calls.
Consensus SNV from all four callers were collected and by default, calls that were detected in at least two calling algorithms or marked with “HotSpotAllele” were retained.</p>
<p>For all SNVs, potential non-hotspot germline variants were removed if they had a normal depth &lt;= 7 and gnomAD allele frequency &gt; 0.001.
Final results were saved in MAF format.</p>
<h4 id="somatic-copy-number-variant-cnv-calling">Somatic Copy Number Variant (CNV) Calling</h4>
<p>We called copy number variants for tumor/normal samples using Control-FREEC <span class="citation" data-cites="ZQ0L3o1q 1F3i4BvCt">[<a href="#ref-ZQ0L3o1q" role="doc-biblioref">11</a>,<a href="#ref-1F3i4BvCt" role="doc-biblioref">12</a>]</span> and CNVkit <span class="citation" data-cites="UTxRcYIQ">[<a href="#ref-UTxRcYIQ" role="doc-biblioref">13</a>]</span> as described in the OpenPBTA manuscript <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
We used GATK <span class="citation" data-cites="14nSa8zB9">[<a href="#ref-14nSa8zB9" role="doc-biblioref">14</a>]</span> to call CNVs for matched tumor/normal WGS samples when there were at least 30 male and 30 female normals from the same sequencing platform available for panel of normal creation.
For tumor only samples, we used Control-FREEC with the following modifications.
Instead of the b-allele frequency germline input file, we used the <code>dbSNP_v153_ucsc-compatible.converted.vt.decomp.norm.common_snps.vcf.gz</code> <a href="s3://kids-first-seq-data/pipeline_references/dbSNP_v153_ucsc-compatible.converted.vt.decomp.norm.common_snps.vcf.gz">dbSNP common snps file</a> and to avoid hard-to-call regions, utilized the <code>hg38_canonical_150.mappability</code> <a href="s3://kids-first-seq-data/pipeline_references/hg38_canonical_150.mappability">mappability file</a>.
Both are also linked in the public <a href="https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-references">Kids First references CAVATICA project</a>.
The Control-FREEC tumor only workflow can be found <a href="https://github.com/kids-first/kf-tumor-workflow/blob/v0.3.0-beta/workflows/kfdrc_controlfreec_tumor_only_wf.cwl">here</a>.</p>
<h4 id="somatic-structural-variant-calling-wgs-samples-only">Somatic Structural Variant Calling (WGS samples only)</h4>
<p>Please refer to the OpenPBTA manuscript for details <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.</p>
<h4 id="methylation-analysis">Methylation Analysis</h4>
<p><strong>Methylation array preprocessing</strong></p>
<p>We preprocessed raw Illumina 450K and EPIC 850K Infinium Human Methylation Bead Array intensities using the array preprocessing methods implemented in the <code>minfi Bioconductor package</code> <span class="citation" data-cites="dxeON3tz">[<a href="#ref-dxeON3tz" role="doc-biblioref">15</a>]</span>.
We utilized either <code>preprocessFunnorm</code> when an array dataset had both tumor and normal samples or multiple OpenPedcan-defined <code>cancer_groups</code> and <code>preprocessQuantile</code> when an array dataset had only tumor samples from a single OpenPedcan-defined <code>cancer_group</code> to estimate usable methylation measurements (<code>beta-values</code> and <code>m-values</code>) and copy number (<code>cn-values</code>).
Some Illumina Infinium array probes targeting CpG loci contain single-nucleotide polymorphisms (SNPs) near or within the probe <span class="citation" data-cites="15Yz3j9AA">[<a href="#ref-15Yz3j9AA" role="doc-biblioref">16</a>]</span>, which could affect DNA methylation measurements <span class="citation" data-cites="HOfcb651">[<a href="#ref-HOfcb651" role="doc-biblioref">17</a>]</span>.
As the minfi preprocessing workflow recommends, we dropped probes containing common SNPs in dbSNP (minor allele frequency &gt; 1%) at the CpG interrogation or the single nucleotide extensions.</p>
<p>Details of methylation array preprocessing are available in the <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/methylation-preprocessing">OpenPedCan methylation-preprocessing module</a>.</p>
<p><strong>Methylation classification of brain tumor molecular subtypes</strong></p>
<p>The Clinical Methylation Unit Laboratory of Pathology at the National Cancer Institute Center for Cancer Research ran the <a href="https://www.molecularneuropathology.org/mnp/classifiers/11">DKFZ brain classifier version 12.6</a>, a comprehensive DNA methylation-based classification of CNS tumors across all entities and age groups <span class="citation" data-cites="19a3Xf4h3">[<a href="#ref-19a3Xf4h3" role="doc-biblioref">18</a>]</span> and/or the Bethesda Brain tumor classifier v2.0 (NIH_v2) and the combo reporter pipeline v2.0 on docker container trust1/bethesda:latest.
Unprocessed IDAT-files from the <a href="https://cbtn.org/">Children’s Brain Tumor Network (CBTN)</a> <code>Infinium Human Methylation EPIC (850k) BeadChip arrays</code> were used as input and the following information was compiled into the <code>histologies.tsv</code> file: <code>dkfz_v12_methylation_subclass</code> (predicted methylation subtype), <code>dkfz_v12_methylation_subclass_score</code> (classification score), <code>dkfz_v12_methylation_mgmt_status</code> (<em>MGMT</em> methylation status), <code>dkfz_v12_methylation_mgmt_estimated</code> (estimated <em>MGMT</em> methylation fraction), <code>NIH_v2_methylation_Superfamily</code>, <code>NIH_v2_methylation_Superfamily_mean_score</code>, <code>NIH_v2_methylation_Superfamily_Consistency_score</code>, <code>NIH_v2_methylation_Class</code>, <code>NIH_v2_methylation_Class_mean_score</code>, <code>NIH_v2_methylation_Class_consistency_score</code>, <code>NIH_v2_methylation_Superfamily_match</code>, and <code>NIH_v2_methylation_Class_match</code>.</p>
<h4 id="gene-expression">Gene Expression</h4>
<p>The <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/tumor-normal-differential-expression"><code>tumor-normal-differential-expression</code></a> module performs differential expression analyses for all sets of Disease (<code>cancer_group</code>) and Dataset (<code>cohort</code>) across all genes found in the <code>gene-expression-rsem-tpm-collapsed.rds</code> table.
The purpose of this analysis is to highlight the correlation and understand the variability in gene expression in different cancer conditions across different histological tissues.
For OpenPedCan v12 data release, this module performs expression analysis over 102 cancer groups across 52 histological tissues for all 54,346 genes found in the dataset.
This analysis was performed on the Children’s Hospital of Philadelphia HPC and was configured to use 96G of RAM per CPU, with one task (one iteration of expression analysis for each set of tissue and cancer group) per CPU (total 102x52=5304 CPUs) using the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">R/DESeq2</a> package.
Please refer to script <code>run-tumor-normal-differential-expression.sh</code> in the module for additional details on Slurm processing configuration.
The same analysis can also be performed on CAVATICA, but requires further optimization.
The module describes the steps for CAVATICA set up, and scripts to publish an application on the portal.
The required data files are also available publicly on CAVATICA under the <a href="https://cavatica.sbgenomics.com/u/cavatica/opentarget/files/#q?path=v12">Open Pediatric Cancer (OpenPedCan) Open Access</a>.
Refer to the module for detailed description and scripts.</p>
<p><strong>Abundance Estimation</strong></p>
<p>Among the data sources used for OpenPedCan, GTEx and TCGA used GENCODE v26 and v36, respectively.
Therefore, the gene symbols had to be harmonized to GENCODE v39 for compatibility with the rest of the dataset.
The liftover process was done via a <a href="https://github.com/d3b-center/D3b-DGD-Collaboration/blob/main/scripts/update_gene_symbols.py">custom script</a>.
The script first constructs an object detailing the gene symbol changes from the <a href="https://https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/archive/monthly/tsv/hgnc_complete_set_2021-06-01.txt">HGNC symbol database</a>.
Using the symbol-change object, the script updates any columns containing gene symbols.
This liftover process was used on GTEx RNA-Seq, TCGA RNA-Seq, DGD fusions, and DNA hotspot files.</p>
<p>Additionally, the gene expression matrices had some instances where multiple Ensembl gene identifiers mapped to the same gene symbol.
This was dealt with by filtering the expression matrix to only genes with [FPKM/TPM] &gt; 0 and then selecting the instance of the gene symbol with the maximum mean [FPKM/TPM/Expected_count] value across samples.
This enabled many downstream modules that require RNA-seq data have gene symbols as unique gene identifiers.
Refer to <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/collapse-rnaseq">collapse-rnaseq</a> module for scripts and details.</p>
<p><strong>Gene fusion detection from RNA-Seq</strong></p>
<p>Gene fusions were called using Arriba <span class="citation" data-cites="DuQpY5dg">[<a href="#ref-DuQpY5dg" role="doc-biblioref">19</a>]</span> and STAR-Fusion <span class="citation" data-cites="rqYlIiAi">[<a href="#ref-rqYlIiAi" role="doc-biblioref">20</a>]</span> as previously reported in OpenPBTA <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
We updated the <a href="https://github.com/d3b-center/annoFusedata"><code>annoFuseData</code> R package</a> to liftover gene symbols to be concordant with VEP v. 105.
Fusions are now filtered with annoFuse <span class="citation" data-cites="10LfPTs0c">[<a href="#ref-10LfPTs0c" role="doc-biblioref">21</a>]</span> upstream and released in <code>fusion-annoFuse.tsv.gz</code>.</p>
<p><strong>Gene fusion detection from fusion panels (DGD only)</strong></p>
<p>Clinical RNA fusion calls from the <a href="https://www.testmenu.com/chop/Tests/785504">CHOP DGD fusion panel</a> are included in the data release in the <code>fusion-dgd.tsv.gz</code> file.</p>
<h4 id="splicing-quantification">Splicing quantification</h4>
<p>To detect alternative splicing events, we utilized rMATS turbo (v. 4.1.0) with Ensembl/GENCODE v39 GFF annotations using the <a href="https://github.com/kids-first/kf-rnaseq-workflow/blob/master/workflow/rmats_wf.cwl">Kids First RNA-Seq workflow</a>.
We used <code>--variable-read-length</code> and <code>-t paired</code> options and applied an additional filter to include only splicing events with total junction read counts greater than 10.</p>
<h4 id="cptac-pbta-cptac-gbm-and-hope-proteogenomics">CPTAC PBTA, CPTAC GBM, and HOPE proteogenomics</h4>
<p>The following methods are the general proteomics approaches used for the CPTAC PBTA <span class="citation" data-cites="meH98mKZ">[<a href="#ref-meH98mKZ" role="doc-biblioref">4</a>]</span>, CPTAC GBM <span class="citation" data-cites="XOVss5Jf">[<a href="#ref-XOVss5Jf" role="doc-biblioref">5</a>]</span>, and HOPE (pre-publication, correspondence with Dr. Pei Wang) studies.
For specific descriptions of sample preparation, mass spectrometry instrumentation and approaches, and data generation, processing, or analysis please refer to the relevant publications.</p>
<p><strong>TMT-11 Labeling and Phosphopeptide Enrichment</strong></p>
<p>Proteome and phosphoproteome analysis of brain cancer samples in the CPTAC PBTA (pediatric), CPTAC GBM (adult), and HOPE (adolescent and young adult, AYA) cohort studies were structured as TMT11-plex experiments.
Tumor samples were digested with LysC and trypsin.
Digested peptides were labeled with TMT11-plex reagent and prepared for phosphopeptide enrichment.
For each dataset, a common reference sample was compiled from representative samples within the cohort.
Phosphopeptides were enriched using Immobilized Metal Affinity Chromatography (IMAC) with Fe3+-NTA-agarose bead kits.</p>
<p><strong>Liquid Chromatography with Tandem Mass Spectrometry (LC-MS/MS) Analysis</strong></p>
<p>To reduce sample complexity, peptide samples were separated by high pH reversed phase HPLC fractionation.
For CPTAC PBTA a total of 96 fractions were consolidated into 12 final fractions for LC-MS/MS analysis.
For CPTAC GBM and HOPE cohorts a total of 96 fractions were consolidated into 24 fractions.
For CPTAC PBTA, global proteome mass spectrometry analyses were performed on an Orbitrap Fusion Tribrid Mass Spectrometer and phosphoproteome analyses were performed on an Orbitrap Fusion Lumos Tribrid Mass Spectrometer.
For CPTAC GBM and HOPE studies, mass spectrometry analysis was performed using an Orbitrap Fusion Lumos Mass Spectrometer.</p>
<p><strong>Protein Identification</strong></p>
<p>The CPTAC PBTA spectra data were analyzed with MSFragger version 20190628 <span class="citation" data-cites="bKpYR6PB">[<a href="#ref-bKpYR6PB" role="doc-biblioref">22</a>]</span> searching against a CPTAC harmonized RefSeq-based sequence database containing 41,457 proteins mapped to the human reference genome (GRCh38/hg38) obtained via the UCSC Table Browser on June 29, 2018, with the addition of 13 proteins encoded in the human mitochondrial genome, 264 common laboratory contaminant proteins, and an equal number of decoy sequences.
The CPTAC GBM and HOPE spectra data were analyzed with MS-GF+ v9881 <span class="citation" data-cites="2rzcFP4J 1B2RVFOKK G4kbj7VC">[<a href="#ref-2rzcFP4J" role="doc-biblioref">23</a>,<a href="#ref-1B2RVFOKK" role="doc-biblioref">24</a>,<a href="#ref-G4kbj7VC" role="doc-biblioref">25</a>]</span> searching against the RefSeq human protein sequence database downloaded on June 29, 2018 (hg38; 41,734 proteins), combined with 264 contaminants, and a decoy database composed of the forward and reversed protein sequences.</p>
<p><strong>Protein Quantification and Data Analysis</strong></p>
<p>Relative protein (gene) abundance was calculated as the ratio of sample abundance to reference abundance using the summed reporter ion intensities from peptides mapped to the respective gene.
For phosphoproteomic datasets, data were not summarized by protein but left at the phosphopeptide level.
Global normalization was performed on the gene-level abundance matrix (log2 ratio) for global proteomic and on the site-level abundance matrix (log2 ratio) for phosphoproteomic data.
The median, log2 relative protein or peptide abundance for each sample was calculated and used to normalize each sample to achieve a common median of 0.
To identify TMT outliers, inter-TMT t-tests were performed for each individual protein or phosphopeptide.
Batch effects were checked using the log2 relative protein or phosphopeptide abundance and corrected using the Combat algorithm <span class="citation" data-cites="VLwxfMaF">[<a href="#ref-VLwxfMaF" role="doc-biblioref">26</a>]</span>.
Imputation was performed after batch effect correction for proteins or phosphopeptides with a missing rate &lt; 50%.
For the phosphopeptide datasets, 440 markers associated with cold-regulated ischemia genes were filtered and removed.</p>
<h3 id="creation-of-openpedcan-analysis-modules">Creation of OpenPedCan Analysis modules</h3>
<h4 id="gene-set-variation-analysis-gene-set-enrichment-analysis-analysis-module">Gene Set Variation Analysis (<code>gene-set-enrichment-analysis</code> analysis module)</h4>
<p>Please refer to the OpenPBTA manuscript for details <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.</p>
<h4 id="fusion-prioritization-fusion_filtering-analysis-module">Fusion prioritization (<code>fusion_filtering</code> analysis module)</h4>
<p>The <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/fusion_filtering"><code>fusion_filtering</code></a> module filters artifacts and annotates fusion calls, with prioritization for oncogenic fusions, for the fusion calls from STAR-Fusion and Arriba.
After artifact filtering, fusions were prioritized and annotated as “putative oncogenic fusions” when at least one gene was a known kinase, oncogene, tumor suppressor, curated transcription factor, on the COSMIC Cancer Gene Census List, or observed in TCGA.
Fusions were retained in this module if they were called by both callers, recurrent or specific to a cancer group, or annotated as a putative oncogenic fusion.
Please refer to the module linked above for more detailed documentation and scripts.</p>
<h4 id="consensus-cnv-calling-wgs-samples-only-copy_number_consensus_call-analysis-modules">Consensus CNV Calling (WGS samples only) (<code>copy_number_consensus_call*</code> analysis modules)</h4>
<p>We adopted the consensus CNV calling described in OpenPBTA manuscript <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span> with minor adjustments.
For each caller and sample with WGS performed, we called CNVs based on consensus among Control-FREEC <span class="citation" data-cites="ZQ0L3o1q 1F3i4BvCt">[<a href="#ref-ZQ0L3o1q" role="doc-biblioref">11</a>,<a href="#ref-1F3i4BvCt" role="doc-biblioref">12</a>]</span>, CNVkit <span class="citation" data-cites="UTxRcYIQ">[<a href="#ref-UTxRcYIQ" role="doc-biblioref">13</a>]</span>, and GATK <span class="citation" data-cites="14nSa8zB9">[<a href="#ref-14nSa8zB9" role="doc-biblioref">14</a>]</span>.
Sample and consensus caller files with more than 2,500 CNVs were removed to de-noise and increase data quality, based on cutoffs used in GISTIC <span class="citation" data-cites="Z38xUgBW">[<a href="#ref-Z38xUgBW" role="doc-biblioref">27</a>]</span>.
For each sample, we included the following regions in the final consensus set: 1) regions with reciprocal overlap of 50% or more between at least two of the callers; 2) smaller CNV regions in which more than 90% of regions were covered by another caller.
For GATK, if a panel of normal was not able to be created (required 30 male and 30 female with the same sequencing platform), consensus was run for that tumor using Control-FREEC, CNVkit, and MantaSV.
We defined copy number as NA for any regions that had a neutral call for the samples included in the consensus file.
We merged CNV regions within 10,000 bp of each other with the same direction of gain or loss into single region.</p>
<p>Any CNVs that overlapped 50% or more with immunoglobulin, telomeric, centromeric, segment duplicated regions, or that were shorter than 3000 bp were filtered out.
The CNVKit calls for WXS samples were appended to the consensus CNV file.</p>
<h4 id="focal-copy-number-calling-focal-cn-file-preparation-analysis-module">Focal Copy Number Calling (<code>focal-cn-file-preparation</code> analysis module)</h4>
<p>Please refer to the OpenPBTA manuscript for details on assignment of copy number status values to CNV segments, cytobands, and genes <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
We applied criteria to resolve instances of multiple conflicting status calls for the same gene and sample, which are described in detail in the <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/focal-cn-file-preparation">focal-cn-file-preparation</a> module.
Briefly, we prioritized 1) non-neutral status calls, 2) calls made from dominant segments with respect to gene overlap, and 3) amplification and deep deletion status calls over gain and loss calls, respectively, when selecting a dominant status call per gene and sample.
These methods resolved &gt;99% of duplicated gene-level status calls.</p>
<h4 id="mutational-signatures-mutational-signatures-analysis-module">Mutational Signatures (<code>mutational-signatures</code> analysis module)</h4>
<p>We obtained mutational signature weights (i.e., exposures) from consensus SNVs using the deconstructSigs R package <span class="citation" data-cites="KhxTOfIb">[<a href="#ref-KhxTOfIb" role="doc-biblioref">28</a>]</span>.
We estimated weights for single- and double-base substitution (SBS and DBS, respectively) signatures from the Catalogue of Somatic Mutations in Cancer (COSMIC) database versions 2 and 3.3, as well as SBS signatures from Alexandrov et al. 2013 <span class="citation" data-cites="kG8qNLrs">[<a href="#ref-kG8qNLrs" role="doc-biblioref">29</a>]</span>.
The following COSMIC SBS signatures were excluded from weight estimation in all tumors: 1) sequencing artifact signatures, 2) signatures associated with environmental exposure, and 3) signatures with an unknown etiology.
Additionally, we excluded therapy-associated signatures from mutational signature weight estimation in tumors collected prior to treatment (i.e. “Initial CNS Tumor” or “Primary Tumor”).</p>
<h4 id="tumor-mutation-burden-tmb-tmb-calculation-analysis-module">Tumor Mutation Burden [TMB] (<code>tmb-calculation</code> analysis module)</h4>
<p>Recent clinical studies have associated high TMB with improved patient response rates and survival benefit from immune checkpoint inhibitors <span class="citation" data-cites="1EbWJlH4p">[<a href="#ref-1EbWJlH4p" role="doc-biblioref">30</a>]</span>.</p>
<p>The <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/tmb-calculation">Tumor Mutation Burden (TMB) <code>tmb-calculation</code></a> module was adapted from the <a href="https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/snv-callers"><code>snv-callers</code> module</a> of the OpenPBTA project <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.
Here, we use mutations in the <code>snv-consensus-plus-hotspots.maf.tsv.gz</code> file which is generated using <a href="https://github.com/kids-first/kf-somatic-workflow/blob/master/docs/kfdrc-consensus-calling.md">Kids First DRC Consensus Calling Workflow</a> and is included in the OpenPedCan data download.
The consensus MAF contains SNVs or MNVs called in at least 2 of the 4 callers (Mutect2, Strelka2, Lancet, and Vardict) plus hotspot mutations if called in 1 of the 4 callers.
We calculated TMB for tumor samples sequenced with either WGS or WXS.
Briefly, we split the SNV consensus MAF into SNVs and multinucleotide variants (MNVs).
We split the MNV subset into SNV calls, merged those back with the SNVs subset, and then removed sample-specific redundant calls.
The resulting merged and non-redundant SNV consensus calls were used as input for the TMB calculation.
We tallied only nonsynonymous variants with classifications of high/moderate consequence (“Missense_Mutation”, “Frame_Shift_Del”, “In_Frame_Ins”, “Frame_Shift_Ins”, “Splice_Site”, “Nonsense_Mutation”, “In_Frame_Del”, “Nonstop_Mutation”, and “Translation_Start_Site”) for the numerator.
All BED files are provided in the data release.</p>
<p><strong>All mutation TMB</strong></p>
<p>For WGS samples, we calculated the size of the genome covered as the intersection of Strelka2 and Mutect2’s effectively surveyed areas, regions common to all variant callers, and used this as the denominator.
<code>WGS_all_mutations_TMB = (total # mutations in consensus MAF) / intersection_strelka_mutect_vardict_genome_size</code>
For WXS samples, we used the size of the WXS bed region file as the denominator.
<code>WXS_all_mutations_TMB = (total # mutations in consensus MAF)) / wxs_genome_size</code></p>
<p><strong>Coding only TMB</strong></p>
<p>We generated coding only TMB from the consensus MAF as well.
We calculated the intersection for Strelka2 and Mutect2 surveyed regions using the coding sequence ranges in the GENCODE v39 gtf supplied in the OpenPedCan data download.
We removed SNVs outside of these coding sequences prior to implementing the TMB calculation below:
<code>WGS_coding_only_TMB = (total # coding mutations in consensus MAF) / intersection_wgs_strelka_mutect_vardict_CDS_genome_size</code>
For WXS samples, we intersected each WXS bed region file with the GENCODE v39 coding sequence, sum only variants within this region for the numerator, and calculate the size of this region as the denominator.
<code>WXS_coding_only_TMB = (total # coding mutations in consensus MAF) / intersection_wxs_CDS_genome_size</code></p>
<p>Finally, we include an option (<code>nonsynfilter_focr</code>) to use specific nonsynonymous mutation variant classifications recommended from the <a href="https://friendsofcancerresearch.org/tmb/">TMB Harmonization Project</a>.</p>
<h4 id="molecular-subtyping">Molecular Subtyping</h4>
<p>Here, we build upon the molecular subtyping performed in OpenPBTA <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span> to align with WHO 2021 subtypes <span class="citation" data-cites="6F5vK3sT">[<a href="#ref-6F5vK3sT" role="doc-biblioref">31</a>]</span>.
Molecular subtypes were generated per tumor event and are listed for each biospecimen in <a href="https://github.com/d3b-center/OpenPedCan-analysis/blob/af9e2bb87e552159309c277ff8fef88c16fe8dcf/tables/results/SuppTable1-Histologies.xlsx"><strong>Supplemental Table S1</strong></a>, with the number of tumors grouped by broad histology and molecular subtype in <a href="https://github.com/d3b-center/OpenPedCan-analysis/blob/af9e2bb87e552159309c277ff8fef88c16fe8dcf/tables/results/SuppTable2-Molecular-Subtype-Table.xlsx"><strong>Supplemental Table S2</strong></a>.</p>
<p><strong>High-grade gliomas</strong></p>
<p>High-grade gliomas (HGG) were categorized based on a combination of clinical information, molecular features, and DNA methylation data.
H3 K28-altered diffuse midline gliomas (DMG) were classified based on the presence of a p.K28M or p.K28I mutation in <em>H3F3A</em>, <em>HIST1H3B</em>, <em>HIST1H3C</em>, or <em>HIST2H3C</em>, or a high-confidence DKFZ methylation score (&gt;=0.8) in the appropriate subclass.
Oligodendroglioma, IDH-mutant tumors were classified based on high-confidence “O_IDH” methylation classifications, and oligosarcoma, IDH-mutant tumors were defined as those with high-confidence “OLIGOSARC_IDH” methylation classifications.
Pleomorphic xanthoastrocytomas (PXA) were classified using the following criteria: 1) methylation subtype is high-confidence “PXA” or <code>pathology_free_text_diagnosis</code> contains “pleomorphic xanthoastrocytoma” or “pxa”, and 2) tumor contains a BRAF V600E mutation and a <em>CDKN2A</em> or <em>CDKN2B</em> homozygous deletion.
Methylation classifications were used in classifying the following subtypes:</p>
<ol type="1">
<li>DHG, H3 G35 (“DHG_G34” and “GBM_G34” classifications)</li>
<li>HGG, IDH (“A_IDH_HG” and “GBM_IDH” classifications)</li>
<li>HGG, H3 wild type (methylation classification contains “GBM_MES”, “GBM_RTK”, “HGG_”, “HGAP”, “AAP”, or “ped_”)</li>
</ol>
<p>A new high-grade glioma entity called infant-type hemispheric gliomas (IHGs), characterized by distinct gene fusions enriched in receptor tyrosine kinase (RTK) genes including <em>ALK</em>, <em>NTRK1/2/3</em>, <em>ROS1</em> or <em>MET</em>, was identified in 2021 <span class="citation" data-cites="gaCrHroW">[<a href="#ref-gaCrHroW" role="doc-biblioref">32</a>]</span>.
To identify IHG tumors, first, tumors which were classified as “IHG” by the DKFZ methylation classifier or diagnosed as “infant type hemispheric glioma” from <code>pathology_free_text_diagnosis</code> were selected <span class="citation" data-cites="19a3Xf4h3">[<a href="#ref-19a3Xf4h3" role="doc-biblioref">18</a>]</span>.
Then, the corresponding tumor RNA-seq data were utilized to seek the evidence for RTK gene fusion.
Based on the specific RTK gene fusion present in the samples, IHGs were further classified as “IHG, ALK-altered”, “IHG, NTRK-altered”, “IHG, ROS1-altered”, or “IHG, MET-altered”.
If no fusion was observed, the samples were identified as “IHG, To be classified”.</p>
<p><strong>Atypical teratoid rhabdoid tumors</strong></p>
<p>Atypical teratoid rhabdoid tumors (ATRT) tumors were categorized into three subtypes: “ATRT, MYC”, “ATRT, SHH”, and “ATRT, TYR” <span class="citation" data-cites="5NHLHVO3">[<a href="#ref-5NHLHVO3" role="doc-biblioref">33</a>]</span>.
In OpenPedCan, the molecular subtyping of ATRT was based solely on the DNA methylation data.
Briefly, ATRT samples with a high confidence DKFZ methylation subclass score (&gt;= 0.8) were selected and subtypes were assigned based on the DKFZ methylation subclass <span class="citation" data-cites="19a3Xf4h3">[<a href="#ref-19a3Xf4h3" role="doc-biblioref">18</a>]</span>.
Samples with low confidence DKFZ methylation subclass scores (&lt; 0.8) were identified as “ATRT, To be classified”.</p>
<p><strong>Neuroblastoma tumors</strong></p>
<p>Neuroblastoma (NBL) tumors with a pathology diagnosis of neuroblastoma, ganglioneuroblastoma, or ganglioneuroma were subtyped based on their MYCN copy number status as either “NBL, MYCN amplified” or “NBL, MYCN non-amplified”.
If <code>pathology_free_text_diagnosis</code> was “NBL, MYCN non-amplified” and the genetic data suggested MYCN amplification, the samples were subtyped as “NBL, MYCN amplified”.
On the other hand, if <code>pathology_free_text_diagnosis</code> was “NBL, MYCN amplified” and the genetic data suggested MYCN non-amplification, the RNA-Seq gene expression level of <em>MYCN</em> was used as a prediction indicator.
In those cases, samples with <em>MYCN</em> gene expression above or below the cutoff (TPM &gt;= 140.83 based on visual inspection of MYCN CNV status) were subtyped as “NBL, MYCN amplified” and “NBL, MYCN non-amplified”, respectively.
<em>MYCN</em> gene expression was also used to subtype samples without DNA sequencing data.
If a sample did not fit none of these situations, it was denoted as “NBL, To be classified”.</p>
<p><strong>Craniopharyngiomas</strong></p>
<p>In addition to molecular criteria established in OpenPBTA <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>, craniopharyngiomas (CRANIO) are now subtyped using DNA methylation classifiers.
Craniopharyngiomas with a high-confidence methylation subclass containing “CPH_PAP” were classified as papillary (CRANIO, PAP), and those with high-confidence methylation subclass containing “CPH_ADM” were classified as adamantinomatous (CRANIO, ADAM), respectively.</p>
<p><strong>Ependymomas</strong></p>
<p>Ependymomas (EPN) are subtyped using the following criteria:</p>
<ol type="1">
<li>Any spinal tumor with <em>MYCN</em> amplification or with a high-confidence “EPN, SP-MYCN” methylation classification was subtyped as EPN, spinal and MYCN-amplified (SP-MYCN).</li>
<li>EPN tumors containing one or more gene fusions of <em>YAP1::MAMLD1</em>, <em>YAP1::MAML2</em>, or <em>YAP1::FAM118B</em>, or else had a high-confidence “EPN, ST YAP1” methylation classification were subtyped as EPN, ST YAP1.</li>
<li>EPN tumors containing one or more gene fusions of <em>ZFTA::RELA</em> or <em>ZFTA::MAML2</em>, or else had a high-confidence “EPN, ST ZFTA” methylation classification were subtyped as EPN, ST ZFTA.
This reflects an update to WHO classifications that now characterizes this subtype based on <em>ZFTA</em> fusions rather than <em>RELA</em> fusions.</li>
<li>EPN tumors with 1) chromosome 1q gain and <em>TKTL1</em> over-expression, or 2) <em>EZHIP</em> over-expression, or 3) posterior fossa anatomical location and a histone H3 K28 mutation in <em>H3F3A</em>, <em>HIST1H3B</em>, <em>HIST1H3C</em>, or <em>HIST2H3C</em>, or 4) a high-confidence “EPN, PF A” methylation classification were subtyped as posterior fossa group A ependymomas (EPN, PF A).</li>
<li>Tumors with 1) chr 6p or 6q loss and <em>GPBP1</em> or <em>IFT46</em> over-expression, or 2) a high-confidence “EPN, PF B” methylation classification were subtyped as posterior fossa group B ependymomas (EPN, PF B).</li>
<li>EPN tumors with a high-confidence “EPN, MPE” methylation classification were subtyped as myxopapillary ependymomas (EPN, MPE).</li>
<li>EPN tumors with a high-confidence “EPN, PF SE” methylation classification were subtyped as posterior fossa subependymomas (EPN, PF SE).</li>
<li>EPN tumors with a high-confidence “EPN, SP SE” methylation classification were subtyped as spinal subependymomas (EPN, SP SE).</li>
<li>EPN tumors with a high-confidence “EPN, SP” methylation classification were subtyped as spinal ependymomas (EPN, SP).</li>
<li>All other EPN tumors were classified as “EPN, To be classified”.</li>
</ol>
<p><strong>Low-grade gliomas</strong></p>
<p>In addition to subtyping methods described in OpenPBTA <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>, high-confidence methylation classifications are now used in classifying the following low-grade glioma (LGG) subtypes:</p>
<ol type="1">
<li>LGG, other MAPK-altered (methylation subclass “PA_MID” or “PLNTY”)</li>
<li>LGG, FGFR-altered (methylation subclass “PA_INF_FGFR”)</li>
<li>LGG, IDH-altered (methylation subclass “A_IDH_LG”)</li>
<li>LGG, MYB/MYBL1 fusion (methylation subclass “AG_MYB” or “LGG_MYB”)</li>
<li>LGG, MAPK-altered (methylation subclass “LGG, MAPK”)</li>
<li>LGG, BRAF- and MAPK-altered (methylation subclass “LGG, BRAF/MAPK”)</li>
<li>SEGA, to be classified (methylation subclass “SEGA, To be classified”)</li>
</ol>
<p><strong>Medulloblastomas (MBs)</strong>
In addition to our previous work classifying MB tumors into the four major subtypes (WNT, SHH, Group 3, and Group 4) using the transcriptomic MedulloClassifier <span class="citation" data-cites="12DKhuCiy">[<a href="#ref-12DKhuCiy" role="doc-biblioref">34</a>]</span>, we integrated high-confidence methylation classification, demographic, and molecular criteria to molecularly subtype SHH tumors into one of four subgroups (alpha, beta, gamma, or delta) (<strong>Figure <a href="#fig:Fig3">3</a></strong>).</p>
<div id="fig:Fig3" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/d3b-center/OpenPedCan-analysis/fc87ad125aaae6d43f17e2baf94551c9a0444bde/figures/manuscript_OPC/figure3/Figure3.png?sanitize=true" style="width:7in" alt="Figure 3: Medulloblastoma Sample Clustering. A, UMAP projection of 271 MB tumors and B, 63 SHH-activated MB tumors using methylation beta values of the 20,000 most variable probes from the Infinium MethylationEPIC array. C, UMAP projection of MB, SHH activated samples indicating copy number status of SHH subgroup known somatic driver genes CCND2, GLI2, MYCN, and PTEN." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>Medulloblastoma Sample Clustering.</strong> A, UMAP projection of 271 MB tumors and B, 63 SHH-activated MB tumors using methylation beta values of the 20,000 most variable probes from the Infinium MethylationEPIC array. C, UMAP projection of MB, SHH activated samples indicating copy number status of SHH subgroup known somatic driver genes <em>CCND2</em>, <em>GLI2</em>, <em>MYCN</em>, and <em>PTEN</em>.</figcaption>
</figure>
</div>
<p>We implemented molecular subtyping as follows:</p>
<ol type="1">
<li>MB tumors with methylation classification that contains “MB_SHH” are subtyped as SHH-activated medulloblastoma (MB, SHH)</li>
<li>MB tumors with “MB_G34_I”, “MB_G34_II”, “MB_G34_III”, and “MB_G334_IV” methylation classifications are subtyped as medulloblastoma group 3 (MB, Group3)</li>
<li>MB tumors with “MB_G34_V”, “MB_G34_VI”, “MB_G34_VII”, and “MB_G334_VIII” methylation classifications are subtyped as medulloblastoma group 4 (MB, Group4)</li>
<li>MB tumors with “MB_WNT” methylation classification are subtyped as WNT-activated MB (MB, WNT)</li>
<li>MB tumors with “MB_MYO” methylation classification are subtyped as medulloblastomas with myogenic differentiation (MB, MYO)</li>
</ol>
<p>We classified MB, SHH subtype tumors using the following criteria:</p>
<ol type="1">
<li><p><em>MB, SHH alpha</em>: sample has a high-confidence “MB_SHH_3” methylation classification, or patient had an age at diagnosis &gt;= 2 years and harbored one of the following molecular alterations in tumor or germline:</p>
<ul>
<li><em>MYCN</em>, <em>GLI2</em>, or <em>CCND2</em> amplification or sample TPM z-score &gt;= 2 in tumor.</li>
<li>A pathogenic or likely pathogenic germline variant in <em>ELP1</em> or <em>TP53</em>.</li>
<li>A <em>TP53</em> hotspot mutation in tumor.</li>
<li>Chromosome 9p gain or chromosome 17p loss in tumor.</li>
</ul></li>
<li><p><em>MB, SHH beta</em>: sample has a high-confidence “MB_SHH_1” methylation classification, or patient had an age at diagnosis &lt; 5 years and harbored one of the following molecular alterations:</p>
<ul>
<li>A <em>KMT2D</em> loss of function variant.</li>
<li><em>PTEN</em> copy number loss or deep deletion, or sample TPM z-score &lt; -2.</li>
<li>Chromosome 2p or 2q gain.</li>
</ul></li>
<li><p><em>MB, SHH gamma</em>: sample has a high-confidence “MB_SHH_2” methylation classification, or patient had an age at diagnosis &lt; 5 years and tumor harbored a chromosome 2p arm gain.</p></li>
<li><p><em>MB, SHH delta</em>: sample has a high-confidence “MB_SHH_4” methylation classification, or patient had an age at diagnosis &gt;= 10 years and harbored one of the following molecular alterations in tumor:</p>
<ul>
<li>a <em>DDX3X</em> or <em>SMO</em> loss-of-function mutation.</li>
<li>a hotspot <em>TERT</em> or U1 snRNA gene mutation.</li>
<li>Chromosome 14q arm loss.</li>
</ul></li>
</ol>
<p><strong>Pineoblastomas</strong></p>
<p>Pineoblastomas (PB) are classified as follows using high-confidence methylation classifications:</p>
<ol type="1">
<li>Pineoblastoma, MYC/FOXR2-activated (“PB_FOXR2” methylation classification)</li>
<li>Pineoblastoma, RB1-altered (“PB_RB1” methylation classification)</li>
<li>Pineoblastoma, group 1 (“PB_GRP1A” and “PB_GRP1B” methylation classifications)</li>
<li>Pineoblastoma, group 2 (“PB_GRP2” methylation classification)</li>
<li>All other pineoblastomas were classified as “PB, To be classified”</li>
</ol>
<p><strong>non-MB, non-ATRT Embryonal Tumors</strong></p>
<p>Updates were made to non-MB, non-ATRT embryonal tumor subtyping as follows:</p>
<ol type="1">
<li>Embryonal tumors with multilayered rosettes and C19MC-altered (ETMR, C19MC-altered) were classified based on 1) high-confidence “ETMR_C19MC” methylation classification or 2) <em>TTYH1</em> gene fusion and either chromosome 19 amplification or <em>LIN28A</em> over-expression.</li>
<li>ETMR, not otherwise specified (NOS) were classified based on <em>LIN28A</em> over-expression and no <em>TTYH1</em> gene fusion.</li>
</ol>
<h4 id="tp53-alteration-annotation-tp53_nf1_score-analysis-module">TP53 Alteration Annotation (<code>tp53_nf1_score</code> analysis module)</h4>
<p>Please refer to the OpenPBTA manuscript for details <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>.</p>
<h4 id="selection-of-independent-samples-independent-samples-analysis-module">Selection of independent samples (<code>independent-samples</code> analysis module)</h4>
<p>For analyses that require all input biospecimens to be independent, we use the OpenPedCan-analysis <a href="https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/d397339d567ddeff17e7a8cdca892f6a9dd2a0ba/analyses/independent-samples">independent-samples</a> module to select only one biospecimen from each input participant.
For each input participant of an analysis, the independent biospecimen is selected based on the analysis-specific filters and preferences for the biospecimen metadata, such as experimental strategy, cancer group, and tumor descriptor.</p>
<h2 id="data-validation-and-quality-control">Data Validation and Quality Control</h2>
<p>We ran NGSCheckMate <span class="citation" data-cites="A4FXW005">[<a href="#ref-A4FXW005" role="doc-biblioref">35</a>]</span> to confirm tumor/normal sample matches as described in the OpenPBTA manuscript <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span> and excluded mismatched samples.
We also ran <code>somalier relate</code> <span class="citation" data-cites="eoDdE9oT">[<a href="#ref-eoDdE9oT" role="doc-biblioref">36</a>]</span> to identify potential mismatched samples.
We required that at least 20M total reads with 50% of RNA-Seq reads mapped to the human reference for samples to be included in analysis.
We required at least 20X coverage for tumor DNA samples to be included in this analysis.</p>
<h2 id="re-use-potential">Re-use potential</h2>
<p>OpenPedCan serves as a community resource whose outputs and/or code can be leveraged directly to ask research questions or serve as an orthogonal validation dataset.
We encourage re-use of the data, ideas and suggestions for improving the data or adding analyses, and/or direct code contributions through a pull-request.
Further, the analysis modules can be run within the project Docker container locally or on EC2 and scaled as the data size increases.</p>
<h2 id="availability-of-source-code-and-requirements">Availability of source code and requirements</h2>
<p>Project name: The Open Pediatric Cancer (OpenPedCan) Project
Project home page: https://github.com/d3b-center/OpenPedCan-analysis
Operating system(s): Platform independent
Programming languages: R, Python, bash
Other requirements: CAVATICA, Docker image at pgc-images.sbgenomics.com/d3b-bixu/openpedcanverse:latest
License: CC-BY 4.0</p>
<p>Primary analyses were performed using Gabriella Miller Kids First pipelines and are listed in the methods section.
Analysis modules were developed within <a href="https://github.com/AlexsLemonade/OpenPBTA-analysis">https://github.com/AlexsLemonade/OpenPBTA-analysis</a> <span class="citation" data-cites="5VXMHJ7N">[<a href="#ref-5VXMHJ7N" role="doc-biblioref">1</a>]</span>, modified based on OpenPBTA, or newly created and can be found within the <a href="https://github.com/d3b-center/OpenPedCan-analysis">https://github.com/d3b-center/OpenPedCan-analysis</a> publicly available repository.</p>
<p>Software versions are documented in <a href="https://github.com/d3b-center/OpenPedCan-analysis/blob/fc87ad125aaae6d43f17e2baf94551c9a0444bde/tables/results/SuppTable3-List_Package_Table.xlsx"><strong>Supplemental Table 3</strong></a>.</p>
<h2 id="data-availability">Data Availability</h2>
<!-- GigaScience requires authors to deposit the dataset(s) supporting the results reported in submitted manuscripts in a publicly-accessible data repository such as GigaDB (see GigaDB database terms of use for complete details). This section should be included when supporting data are available and must include the name of the repository and the permanent identifier or accession number and persistent hyperlinks for the datasets (if appropriate). The following format is recommended:

"The dataset(s) supporting the results of this article is(are) available in the [repository name] repository, [cite unique persistent identifier]."

Following the Joint Declaration of Data Citation Principles, where appropriate we ask that the data sets be cited where it is first mentioned in the manuscript, and included in the reference list. If a DOI has been issued to a dataset please always cite it using the DOI rather than the less stable URL the DOI resolves to (e.g., http://dx.doi.org/10.5524/100044 rather than http://gigadb.org/dataset/100044). For more see:

A list of available scientific research data repositories can be found in res3data and FAIRsharing.

A Data availability statement is required for all papers, and the code snapshots should be cited here. -->
<h3 id="datasets">Datasets</h3>
<p>The datasets supporting this study are available as follows:
The TARGET dataset is available in dbGAP under phs000218.v23.p8 <span class="citation" data-cites="17Erd7F9J">[<a href="#ref-17Erd7F9J" role="doc-biblioref">37</a>]</span>.
The GMKF Neuroblastoma dataset is available in dbGAP under phs001436.v1.p1<span class="citation" data-cites="UVwAVvuW">[<a href="#ref-UVwAVvuW" role="doc-biblioref">38</a>]</span>.
The Pediatric Brain Tumor Atlas data (PBTA), containing the subcohorts OpenPBTA, Kids First PBTA (X01), Chordoma Foundation, MI-ONCOSEQ Study, PNOC, and DGD is available in dbGAP under phs002517.v2.p2 <span class="citation" data-cites="16JieTeo0">[<a href="#ref-16JieTeo0" role="doc-biblioref">39</a>]</span> or in the Kids First Portal (kidsfirstdrc.org).
The raw Genotype-Tissue Expression (GTEx) dataset is available in dbGAP under phs000424.v9.p2 and publicly available at https://gtexportal.org/home/.
The Cancer Genome Atlas (TCGA) dataset is available in dbGAP under phs000178.v11.p8 <span class="citation" data-cites="A9rC7i7X">[<a href="#ref-A9rC7i7X" role="doc-biblioref">40</a>]</span>.</p>
<p>Merged summary files for the latest release of OpenPedCan are openly accessible in <a href="https://cavatica.sbgenomics.com/u/cavatica/opentarget">CAVATICA</a> or via <code>download-data.sh</code> script in the <a href="https://github.com/d3b-center/OpenPedCan-analysis">https://github.com/d3b-center/OpenPedCan-analysis</a> repository.
Cancer group summary data from release v11 are visible within the NCI’s pediatric <a href="https://moleculartargets.ccdi.cancer.gov/">Molecular Targets Platform</a>.
Cohort, cancer group, and individual data are visible within <a href="https://pedcbioportal.kidsfirstdrc.org/study/summary?id=openpedcan_v15">PedcBioPortal</a></p>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>We are incredibly grateful to each patient and family for donating tissue and associated metadata and clinical data to their respective consortia.
This project has been funded in whole or in part with Federal funds from the National Cancer Institute, National Institutes of Health, under Contract No. 75N91019D00024, Task Order No. 75N91020F00003 (DMT, JLR, SJD, JMM, ST, AF, ACR).
The content of this publication does not necessarily reflect the views or policies of the Department of Health and Human Services, nor does mention of trade names, commercial products or organizations imply endorsement by the U.S. Government.
We thank Rocky Breslow for GitHub actions contributions and Rust Turakulov for contributing to methylation data analysis.</p>
<h2 id="author-contributions">Author Contributions</h2>
<table>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="header">
<th>Author</th>
<th>Contributions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Zhuangzhuang Geng</td>
<td>Data curation, Formal analysis, Investigation, Methodology, Software, Writing – Original draft</td>
</tr>
<tr class="even">
<td>Eric Wafula</td>
<td>Formal analysis, Software, Investigation, Writing – Original draft</td>
</tr>
<tr class="odd">
<td>Ryan J. Corbett</td>
<td>Formal analysis, Writing - original draft</td>
</tr>
<tr class="even">
<td>Yuanchao Zhang</td>
<td>Software, Formal analysis, Methodology, Writing – Original draft</td>
</tr>
<tr class="odd">
<td>Run Jin</td>
<td>Formal analysis</td>
</tr>
<tr class="even">
<td>Krutika S. Gaonkar</td>
<td>Data curation, Formal analysis, Investigation</td>
</tr>
<tr class="odd">
<td>Sangeeta Shukla</td>
<td>Formal analysis, Investigation, Methodology, Writing – Original draft, Writing - Review and editing</td>
</tr>
<tr class="even">
<td>Komal S. Rathi</td>
<td>Formal analysis, Investigation, Methodology</td>
</tr>
<tr class="odd">
<td>Dave Hill</td>
<td>Formal analysis, Writing - original draft</td>
</tr>
<tr class="even">
<td>Aditya Lahiri</td>
<td>Formal analysis, Investigation, Methodology, Writing – Original draft</td>
</tr>
<tr class="odd">
<td>Daniel P. Miller</td>
<td>Formal analysis, Writing – Original draft</td>
</tr>
<tr class="even">
<td>Alex Sickler</td>
<td>Methodology, Formal analysis</td>
</tr>
<tr class="odd">
<td>Kelsey Keith</td>
<td>Writing - original draft, Formal Analysis</td>
</tr>
<tr class="even">
<td>Christopher Blackden</td>
<td>Software</td>
</tr>
<tr class="odd">
<td>Antonia Chroni</td>
<td>Validation</td>
</tr>
<tr class="even">
<td>Miguel A. Brown</td>
<td>Data curation, Methodology, Formal analysis, Investigation, Software, Supervision, Writing – Original draft</td>
</tr>
<tr class="odd">
<td>Adam A. Kraya</td>
<td>Methodology</td>
</tr>
<tr class="even">
<td>Carl J. Koschmann</td>
<td>Data Curation</td>
</tr>
<tr class="odd">
<td>Kenneth Aldape</td>
<td>Formal Analysis, Investigation, Data curation</td>
</tr>
<tr class="even">
<td>Xiaoyan Huang</td>
<td>Formal analysis, Software</td>
</tr>
<tr class="odd">
<td>Brian R. Rood</td>
<td>Data Curation</td>
</tr>
<tr class="even">
<td>Jennifer L. Mason</td>
<td>Supervision</td>
</tr>
<tr class="odd">
<td>Gerri R. Trooskin</td>
<td>Funding acquisition</td>
</tr>
<tr class="even">
<td>Zied Abdullaev</td>
<td>Formal Analysis, Investigation, Data curation</td>
</tr>
<tr class="odd">
<td>Pei Wang</td>
<td>Formal Analysis, Supervision</td>
</tr>
<tr class="even">
<td>Yuankun Zhu</td>
<td>Supervision</td>
</tr>
<tr class="odd">
<td>Bailey K. Farrow</td>
<td>Data curation, Software, Project administration, Supervision</td>
</tr>
<tr class="even">
<td>Alvin Farrel</td>
<td>Supervision, Investigation, Methodology, Funding acquisition</td>
</tr>
<tr class="odd">
<td>Joseph M. Dybas</td>
<td>Writing – Original draft, Methodology</td>
</tr>
<tr class="even">
<td>Chuwei Zhong</td>
<td>Formal analysis</td>
</tr>
<tr class="odd">
<td>Nicholas Van Kuren</td>
<td>Data curation, Software</td>
</tr>
<tr class="even">
<td>Bo Zhang</td>
<td>Data curation, Formal analysis</td>
</tr>
<tr class="odd">
<td>Mariarita Santi</td>
<td>Investigation, Validation</td>
</tr>
<tr class="even">
<td>Saksham Phul</td>
<td>Formal analysis</td>
</tr>
<tr class="odd">
<td>Asif T Chinwalla</td>
<td>Project administration, Supervision, Methodology, Investigation, Validation</td>
</tr>
<tr class="even">
<td>Adam C. Resnick</td>
<td>Funding acquisition, Resources</td>
</tr>
<tr class="odd">
<td>Sharon J. Diskin</td>
<td>Funding acquisition</td>
</tr>
<tr class="even">
<td>Sarah Tasian</td>
<td>Funding acquisition</td>
</tr>
<tr class="odd">
<td>Stephanie Stefankiewicz</td>
<td>Project administration</td>
</tr>
<tr class="even">
<td>John M. Maris</td>
<td>Funding acquisition</td>
</tr>
<tr class="odd">
<td>Brian M. Ennis</td>
<td>Formal analysis</td>
</tr>
<tr class="even">
<td>Matthew R. Lueder</td>
<td>Data curation</td>
</tr>
<tr class="odd">
<td>Ammar S. Naqvi</td>
<td>Methodology, Writing – Original draft</td>
</tr>
<tr class="even">
<td>Noel Coleman</td>
<td>Data curation</td>
</tr>
<tr class="odd">
<td>Weiping Ma</td>
<td>Formal Analysis, Investigation, Data curation</td>
</tr>
<tr class="even">
<td>Deanne Taylor</td>
<td>Conceptualization, Data curation, Funding acquisition, Investigation, Methodology, Supervision, Project administration</td>
</tr>
<tr class="odd">
<td>Jo Lynne Rokita</td>
<td>Conceptualization, Data curation, Formal analysis, Funding acquisition, Project administration, Investigation, Methodology, Software, Supervision, Writing – Original draft, Writing - Review and editing</td>
</tr>
</tbody>
</table>
<h2 id="declarations-of-interest">Declarations of Interest</h2>
<p>The authors declare no conflicts.</p>
<!--
## Figure Titles and Legends
-->
<!--
    ## List of Abbreviations -->
<!-- If abbreviations are used in the text they should be defined in the text at first use, and a list of abbreviations should be provided in alphabetical order. -->
<h2 id="supplemental-information-titles-and-legends">Supplemental Information Titles and Legends</h2>
<p><strong>Supplemental Table 1</strong>
README, metadata, and clinical data for each patient and biospecimen in OpenPedCan.</p>
<p><strong>Supplemental Table 2</strong>
Number of tumors and corresponding patients from which WHO 2021 molecular subtypes were generated through OpenPedCan analysis modules are listed in Sheet 1.
Molecular subgroups (alpha, beta, gamma, or delta) for medulloblastoma SHH tumors are listed in Sheet 2.</p>
<p><strong>Supplemental Table 3</strong>
Listed are the software versions for all packages and workflows used in this manuscript.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-5VXMHJ7N" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>OpenPBTA: The Open Pediatric Brain Tumor Atlas</strong> <div class="csl-block">Joshua A Shapiro, Krutika S Gaonkar, Stephanie J Spielman, Candace L Savonen, Chante J Bethell, Run Jin, Komal S Rathi, Yuankun Zhu, Laura E Egolf, Bailey K Farrow, … Jaclyn N Taroni</div> <em>Cell Genomics</em> (2023-07) <a href="https://doi.org/gr92p6">https://doi.org/gr92p6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.xgen.2023.100340">10.1016/j.xgen.2023.100340</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37492101">37492101</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10363844">PMC10363844</a></div></div>
</div>
<div id="ref-SDwYl8uA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Genomic Profiling of Childhood Tumor Patient-Derived Xenograft Models to Enable Rational Clinical Trial Design</strong> <div class="csl-block">Jo Lynne Rokita, Komal S Rathi, Maria F Cardenas, Kristen A Upton, Joy Jayaseelan, Katherine L Cross, Jacob Pfeil, Laura E Egolf, Gregory P Way, Alvin Farrel, … John M Maris</div> <em>Cell Reports</em> (2019-11) <a href="https://doi.org/gg596n">https://doi.org/gg596n</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.celrep.2019.09.071">10.1016/j.celrep.2019.09.071</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31693904">31693904</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6880934">PMC6880934</a></div></div>
</div>
<div id="ref-ygVj7a6q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Michigan Center for Translational Pathology</strong> <a href="https://mctp.med.umich.edu">https://mctp.med.umich.edu</a></div>
</div>
<div id="ref-meH98mKZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Integrated Proteogenomic Characterization across Major Histological Types of Pediatric Brain Cancer</strong> <div class="csl-block">Francesca Petralia, Nicole Tignor, Boris Reva, Mateusz Koptyra, Shrabanti Chowdhury, Dmitry Rykunov, Azra Krek, Weiping Ma, Yuankun Zhu, Jiayi Ji, … William E Bocik</div> <em>Cell</em> (2020-12) <a href="https://doi.org/ghqjkz">https://doi.org/ghqjkz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2020.10.044">10.1016/j.cell.2020.10.044</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33242424">33242424</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8143193">PMC8143193</a></div></div>
</div>
<div id="ref-XOVss5Jf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Proteogenomic and metabolomic characterization of human glioblastoma</strong> <div class="csl-block">Liang-Bo Wang, Alla Karpova, Marina A Gritsenko, Jennifer E Kyle, Song Cao, Yize Li, Dmitry Rykunov, Antonio Colaprico, Joseph H Rothstein, Runyu Hong, … Jun Zhu</div> <em>Cancer Cell</em> (2021-04) <a href="https://doi.org/gh7whf">https://doi.org/gh7whf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ccell.2021.01.006">10.1016/j.ccell.2021.01.006</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33577785">33577785</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8044053">PMC8044053</a></div></div>
</div>
<div id="ref-p1f5DxRQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>The Ensembl Variant Effect Predictor</strong> <div class="csl-block">William McLaren, Laurent Gil, Sarah E Hunt, Harpreet Singh Riat, Graham RS Ritchie, Anja Thormann, Paul Flicek, Fiona Cunningham</div> <em>Genome Biology</em> (2016-06-06) <a href="https://doi.org/gdz75c">https://doi.org/gdz75c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-016-0974-4">10.1186/s13059-016-0974-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27268795">27268795</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4893825">PMC4893825</a></div></div>
</div>
<div id="ref-REfkDUtE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Strelka2: fast and accurate calling of germline and somatic variants</strong> <div class="csl-block">Sangtae Kim, Konrad Scheffler, Aaron L Halpern, Mitchell A Bekritsky, Eunho Noh, Morten Källberg, Xiaoyu Chen, Yeonbin Kim, Doruk Beyter, Peter Krusche, Christopher T Saunders</div> <em>Nature Methods</em> (2018-07-16) <a href="https://doi.org/gdwrp4">https://doi.org/gdwrp4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-018-0051-x">10.1038/s41592-018-0051-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30013048">30013048</a></div></div>
</div>
<div id="ref-149BEKISi" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Calling Somatic SNVs and Indels with Mutect2</strong> <div class="csl-block">David Benjamin, Takuto Sato, Kristian Cibulskis, Gad Getz, Chip Stewart, Lee Lichtenstein</div> <em>Cold Spring Harbor Laboratory</em> (2019-12-02) <a href="https://doi.org/ggntwv">https://doi.org/ggntwv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/861054">10.1101/861054</a></div></div>
</div>
<div id="ref-V6KdWVYi" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Genome-wide somatic variant calling using localized colored de Bruijn graphs</strong> <div class="csl-block">Giuseppe Narzisi, André Corvelo, Kanika Arora, Ewa A Bergmann, Minita Shah, Rajeeva Musunuri, Anne-Katrin Emde, Nicolas Robine, Vladimir Vacic, Michael C Zody</div> <em>Communications Biology</em> (2018-03-22) <a href="https://doi.org/gfcfr8">https://doi.org/gfcfr8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s42003-018-0023-9">10.1038/s42003-018-0023-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30271907">30271907</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6123722">PMC6123722</a></div></div>
</div>
<div id="ref-trQRR8fs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>VarDict: a novel and versatile variant caller for next-generation sequencing in cancer research</strong> <div class="csl-block">Zhongwu Lai, Aleksandra Markovets, Miika Ahdesmaki, Brad Chapman, Oliver Hofmann, Robert McEwen, Justin Johnson, Brian Dougherty, JCarl Barrett, Jonathan R Dry</div> <em>Nucleic Acids Research</em> (2016-04-07) <a href="https://doi.org/f8v6qz">https://doi.org/f8v6qz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkw227">10.1093/nar/gkw227</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27060149">27060149</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4914105">PMC4914105</a></div></div>
</div>
<div id="ref-ZQ0L3o1q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Control-FREEC: a tool for assessing copy number and allelic content using next-generation sequencing data</strong> <div class="csl-block">Valentina Boeva, Tatiana Popova, Kevin Bleakley, Pierre Chiche, Julie Cappo, Gudrun Schleiermacher, Isabelle Janoueix-Lerosey, Olivier Delattre, Emmanuel Barillot</div> <em>Bioinformatics</em> (2011-12-06) <a href="https://doi.org/ckt4vz">https://doi.org/ckt4vz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btr670">10.1093/bioinformatics/btr670</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22155870">22155870</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3268243">PMC3268243</a></div></div>
</div>
<div id="ref-1F3i4BvCt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Control-free calling of copy number alterations in deep-sequencing data using GC-content normalization</strong> <div class="csl-block">Valentina Boeva, Andrei Zinovyev, Kevin Bleakley, Jean-Philippe Vert, Isabelle Janoueix-Lerosey, Olivier Delattre, Emmanuel Barillot</div> <em>Bioinformatics</em> (2010-11-15) <a href="https://doi.org/c6bcps">https://doi.org/c6bcps</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btq635">10.1093/bioinformatics/btq635</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21081509">21081509</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3018818">PMC3018818</a></div></div>
</div>
<div id="ref-UTxRcYIQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>CNVkit: Genome-Wide Copy Number Detection and Visualization from Targeted DNA Sequencing</strong> <div class="csl-block">Eric Talevich, AHunter Shain, Thomas Botton, Boris C Bastian</div> <em>PLOS Computational Biology</em> (2016-04-21) <a href="https://doi.org/c9pd">https://doi.org/c9pd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004873">10.1371/journal.pcbi.1004873</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27100738">27100738</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4839673">PMC4839673</a></div></div>
</div>
<div id="ref-14nSa8zB9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>The Genome Analysis Toolkit: A MapReduce framework for analyzing next-generation DNA sequencing data</strong> <div class="csl-block">Aaron McKenna, Matthew Hanna, Eric Banks, Andrey Sivachenko, Kristian Cibulskis, Andrew Kernytsky, Kiran Garimella, David Altshuler, Stacey Gabriel, Mark Daly, Mark A DePristo</div> <em>Genome Research</em> (2010-07-19) <a href="https://doi.org/bnzbn6">https://doi.org/bnzbn6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/gr.107524.110">10.1101/gr.107524.110</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20644199">20644199</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928508">PMC2928508</a></div></div>
</div>
<div id="ref-dxeON3tz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Preprocessing, normalization and integration of the Illumina HumanMethylationEPIC array with minfi</strong> <div class="csl-block">Jean-Philippe Fortin, Timothy J Triche Jr, Kasper D Hansen</div> <em>Bioinformatics</em> (2016-11-29) <a href="https://doi.org/f9x7kd">https://doi.org/f9x7kd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btw691">10.1093/bioinformatics/btw691</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28035024">28035024</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5408810">PMC5408810</a></div></div>
</div>
<div id="ref-15Yz3j9AA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Review of processing and analysis methods for DNA methylation array data</strong> <div class="csl-block">CS Wilhelm-Benartzi, DC Koestler, MR Karagas, JM Flanagan, BC Christensen, KT Kelsey, CJ Marsit, EA Houseman, R Brown</div> <em>British Journal of Cancer</em> (2013-08-27) <a href="https://doi.org/gb9qvv">https://doi.org/gb9qvv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/bjc.2013.496">10.1038/bjc.2013.496</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23982603">23982603</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3777004">PMC3777004</a></div></div>
</div>
<div id="ref-HOfcb651" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Impact of SNPs on methylation readouts by Illumina Infinium HumanMethylation450 BeadChip Array: implications for comparative population studies</strong> <div class="csl-block">Patrycja Daca-Roszak, Aleksandra Pfeifer, Jadwiga Żebracka-Gala, Dagmara Rusinek, Aleksandra Szybińska, Barbara Jarząb, Michał Witt, Ewa Ziętkiewicz</div> <em>BMC Genomics</em> (2015-11-25) <a href="https://doi.org/gb3h5r">https://doi.org/gb3h5r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-015-2202-0">10.1186/s12864-015-2202-0</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26607064">26607064</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4659175">PMC4659175</a></div></div>
</div>
<div id="ref-19a3Xf4h3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>DNA methylation-based classification of central nervous system tumours</strong> <div class="csl-block">David Capper, David TW Jones, Martin Sill, Volker Hovestadt, Daniel Schrimpf, Dominik Sturm, Christian Koelsche, Felix Sahm, Lukas Chavez, David E Reuss, … Stefan M Pfister</div> <em>Nature</em> (2018-03-14) <a href="https://doi.org/gc5t36">https://doi.org/gc5t36</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature26000">10.1038/nature26000</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29539639">29539639</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6093218">PMC6093218</a></div></div>
</div>
<div id="ref-DuQpY5dg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Accurate and efficient detection of gene fusions from RNA sequencing data</strong> <div class="csl-block">Sebastian Uhrig, Julia Ellermann, Tatjana Walther, Pauline Burkhardt, Martina Fröhlich, Barbara Hutter, Umut H Toprak, Olaf Neumann, Albrecht Stenzinger, Claudia Scholl, … Benedikt Brors</div> <em>Genome Research</em> (2021-01-13) <a href="https://doi.org/gjvdvp">https://doi.org/gjvdvp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/gr.257246.119">10.1101/gr.257246.119</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33441414">33441414</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7919457">PMC7919457</a></div></div>
</div>
<div id="ref-rqYlIiAi" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>STAR-Fusion: Fast and Accurate Fusion Transcript Detection from RNA-Seq</strong> <div class="csl-block">Brian J Haas, Alex Dobin, Nicolas Stransky, Bo Li, Xiao Yang, Timothy Tickle, Asma Bankapur, Carrie Ganote, Thomas G Doak, Nathalie Pochet, … Aviv Regev</div> <em>Cold Spring Harbor Laboratory</em> (2017-03-24) <a href="https://doi.org/gf5pc5">https://doi.org/gf5pc5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/120295">10.1101/120295</a></div></div>
</div>
<div id="ref-10LfPTs0c" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>annoFuse: an R Package to annotate, prioritize, and interactively explore putative oncogenic RNA fusions</strong> <div class="csl-block">Krutika S Gaonkar, Federico Marini, Komal S Rathi, Payal Jain, Yuankun Zhu, Nicholas A Chimicles, Miguel A Brown, Ammar S Naqvi, Bo Zhang, Phillip B Storm, … Jo Lynne Rokita</div> <em>BMC Bioinformatics</em> (2020-12) <a href="https://doi.org/gm84mh">https://doi.org/gm84mh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-020-03922-7">10.1186/s12859-020-03922-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33317447">33317447</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7737294">PMC7737294</a></div></div>
</div>
<div id="ref-bKpYR6PB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>MSFragger: ultrafast and comprehensive peptide identification in mass spectrometry–based proteomics</strong> <div class="csl-block">Andy T Kong, Felipe V Leprevost, Dmitry M Avtonomov, Dattatreya Mellacheruvu, Alexey I Nesvizhskii</div> <em>Nature Methods</em> (2017-04-10) <a href="https://doi.org/f9z6p7">https://doi.org/f9z6p7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.4256">10.1038/nmeth.4256</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28394336">28394336</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5409104">PMC5409104</a></div></div>
</div>
<div id="ref-2rzcFP4J" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Correcting systematic bias and instrument measurement drift with mzRefinery</strong> <div class="csl-block">Bryson C Gibbons, Matthew C Chambers, Matthew E Monroe, David L Tabb, Samuel H Payne</div> <em>Bioinformatics</em> (2015-08-04) <a href="https://doi.org/gb5g57">https://doi.org/gb5g57</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btv437">10.1093/bioinformatics/btv437</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26243018">26243018</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4653383">PMC4653383</a></div></div>
</div>
<div id="ref-1B2RVFOKK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>MS-GF+ makes progress towards a universal database search tool for proteomics</strong> <div class="csl-block">Sangtae Kim, Pavel A Pevzner</div> <em>Nature Communications</em> (2014-10-31) <a href="https://doi.org/ggkdq8">https://doi.org/ggkdq8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ncomms6277">10.1038/ncomms6277</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25358478">25358478</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5036525">PMC5036525</a></div></div>
</div>
<div id="ref-G4kbj7VC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Spectral probabilities of top-down tandem mass spectra</strong> <div class="csl-block">Xiaowen Liu, Matthew W Segar, Shuai Cheng Li, Sangtae Kim</div> <em>BMC Genomics</em> (2014-01) <a href="https://doi.org/gb3gzt">https://doi.org/gb3gzt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2164-15-s1-s9">10.1186/1471-2164-15-s1-s9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24564718">24564718</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4046700">PMC4046700</a></div></div>
</div>
<div id="ref-VLwxfMaF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>A probability-based approach for high-throughput protein phosphorylation analysis and site localization</strong> <div class="csl-block">Sean A Beausoleil, Judit Villén, Scott A Gerber, John Rush, Steven P Gygi</div> <em>Nature Biotechnology</em> (2006-09-10) <a href="https://doi.org/dbwqf4">https://doi.org/dbwqf4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt1240">10.1038/nbt1240</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16964243">16964243</a></div></div>
</div>
<div id="ref-Z38xUgBW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>GISTIC2.0 facilitates sensitive and confident localization of the targets of focal somatic copy-number alteration in human cancers</strong> <div class="csl-block">Craig H Mermel, Steven E Schumacher, Barbara Hill, Matthew L Meyerson, Rameen Beroukhim, Gad Getz</div> <em>Genome Biology</em> (2011-04-28) <a href="https://doi.org/10.1186/gb-2011-12-4-r41">https://doi.org/10.1186/gb-2011-12-4-r41</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/gb-2011-12-4-r41">10.1186/gb-2011-12-4-r41</a></div></div>
</div>
<div id="ref-KhxTOfIb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>deconstructSigs: delineating mutational processes in single tumors distinguishes DNA repair deficiencies and patterns of carcinoma evolution</strong> <div class="csl-block">Rachel Rosenthal, Nicholas McGranahan, Javier Herrero, Barry S Taylor, Charles Swanton</div> <em>Genome Biology</em> (2016-02-22) <a href="https://doi.org/f8bdsq">https://doi.org/f8bdsq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-016-0893-4">10.1186/s13059-016-0893-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26899170">26899170</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4762164">PMC4762164</a></div></div>
</div>
<div id="ref-kG8qNLrs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Signatures of mutational processes in human cancer</strong><div class="csl-block">, Ludmil B Alexandrov, Serena Nik-Zainal, David C Wedge, Samuel AJR Aparicio, Sam Behjati, Andrew V Biankin, … Michael R Stratton</div> <em>Nature</em> (2013-08-14) <a href="https://doi.org/f22m2q">https://doi.org/f22m2q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature12477">10.1038/nature12477</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23945592">23945592</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3776390">PMC3776390</a></div></div>
</div>
<div id="ref-1EbWJlH4p" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Tumor mutational burden standardization initiatives: Recommendations for consistent tumor mutational burden assessment in clinical samples to guide immunotherapy treatment decisions</strong> <div class="csl-block">Albrecht Stenzinger, Jeffrey D Allen, Jörg Maas, Mark D Stewart, Diana M Merino, Madison M Wempe, Manfred Dietel</div> <em>Genes, Chromosomes and Cancer</em> (2019-03-07) <a href="https://doi.org/ggc8pj">https://doi.org/ggc8pj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/gcc.22733">10.1002/gcc.22733</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30664300">30664300</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6618007">PMC6618007</a></div></div>
</div>
<div id="ref-6F5vK3sT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>The 2021 WHO Classification of Tumors of the Central Nervous System: a summary</strong> <div class="csl-block">David N Louis, Arie Perry, Pieter Wesseling, Daniel J Brat, Ian A Cree, Dominique Figarella-Branger, Cynthia Hawkins, HK Ng, Stefan M Pfister, Guido Reifenberger, … David W Ellison</div> <em>Neuro-Oncology</em> (2021-06-29) <a href="https://doi.org/gmqhbf">https://doi.org/gmqhbf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/neuonc/noab106">10.1093/neuonc/noab106</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34185076">34185076</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8328013">PMC8328013</a></div></div>
</div>
<div id="ref-gaCrHroW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Alterations in ALK/ROS1/NTRK/MET drive a group of infantile hemispheric gliomas</strong> <div class="csl-block">Ana S Guerreiro Stucklin, Scott Ryall, Kohei Fukuoka, Michal Zapotocky, Alvaro Lassaletta, Christopher Li, Taylor Bridge, Byungjin Kim, Anthony Arnoldo, Paul E Kowalski, … Cynthia Hawkins</div> <em>Nature Communications</em> (2019-09-25) <a href="https://doi.org/gh7bg8">https://doi.org/gh7bg8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-019-12187-5">10.1038/s41467-019-12187-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31554817">31554817</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6761184">PMC6761184</a></div></div>
</div>
<div id="ref-5NHLHVO3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Molecular subgrouping of atypical teratoid/rhabdoid tumors—a reinvestigation and current consensus</strong> <div class="csl-block">Ben Ho, Pascal D Johann, Yura Grabovska, Mamy Jean De Dieu Andrianteranagna, Fupan Yao, Michael Frühwald, Martin Hasselblatt, Franck Bourdeaut, Daniel Williamson, Annie Huang, Marcel Kool</div> <em>Neuro-Oncology</em> (2019-12-31) <a href="https://doi.org/gn3kcm">https://doi.org/gn3kcm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/neuonc/noz235">10.1093/neuonc/noz235</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31889194">31889194</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7229260">PMC7229260</a></div></div>
</div>
<div id="ref-12DKhuCiy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>A transcriptome-based classifier to determine molecular subtypes in medulloblastoma</strong> <div class="csl-block">Komal S Rathi, Sherjeel Arif, Mateusz Koptyra, Ammar S Naqvi, Deanne M Taylor, Phillip B Storm, Adam C Resnick, Jo Lynne Rokita, Pichai Raman</div> <em>PLOS Computational Biology</em> (2020-10-29) <a href="https://doi.org/gm84kq">https://doi.org/gm84kq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1008263">10.1371/journal.pcbi.1008263</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33119584">33119584</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7654754">PMC7654754</a></div></div>
</div>
<div id="ref-A4FXW005" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>NGSCheckMate: software for validating sample identity in next-generation sequencing studies within and across data types</strong> <div class="csl-block">Sejoon Lee, Soohyun Lee, Scott Ouellette, Woong-Yang Park, Eunjung A Lee, Peter J Park</div> <em>Nucleic Acids Research</em> (2017-03-23) <a href="https://doi.org/f9xrq4">https://doi.org/f9xrq4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkx193">10.1093/nar/gkx193</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28369524">28369524</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5499645">PMC5499645</a></div></div>
</div>
<div id="ref-eoDdE9oT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Somalier: rapid relatedness estimation for cancer and germline studies using efficient genome sketches</strong> <div class="csl-block">Brent S Pedersen, Preetida J Bhetariya, Joe Brown, Stephanie N Kravitz, Gabor Marth, Randy L Jensen, Mary P Bronner, Hunter R Underhill, Aaron R Quinlan</div> <em>Genome Medicine</em> (2020-07-14) <a href="https://doi.org/gtsm62">https://doi.org/gtsm62</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-020-00761-2">10.1186/s13073-020-00761-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32664994">32664994</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7362544">PMC7362544</a></div></div>
</div>
<div id="ref-17Erd7F9J" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>dbGaP Study</strong> <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000218.v23.p8">https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000218.v23.p8</a></div>
</div>
<div id="ref-UVwAVvuW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001436.v1.p1">https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001436.v1.p1</a></div>
</div>
<div id="ref-16JieTeo0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>dbGaP Study</strong> <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs002517.v2.p2">https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs002517.v2.p2</a></div>
</div>
<div id="ref-A9rC7i7X" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>dbGaP Study</strong> <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000178.v11.p8">https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000178.v11.p8</a></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
